<!DOCTYPE html> <head> <meta charset=utf-8> <meta content="width=device-width,initial-scale=1" name=viewport> <meta content=#333333 name=theme-color> <base href=/ > <link href=global.css rel=stylesheet> <style>main.svelte-1pnr8ds{position:relative;box-sizing:border-box;min-height:100vh;width:100%;padding-top:3.5rem;padding-bottom:2.75rem}main.homepage.svelte-1pnr8ds{padding-right:2rem;padding-left:2rem}.bg.contentPage.svelte-1pnr8ds{padding-left:1rem;padding-right:1rem}@media screen and (min-width:775px){.bg.contentPage.svelte-1pnr8ds{padding-left:calc(75px + 1rem);padding-right:4rem}}.bg.svelte-1pnr8ds{background-size:auto,0;background-repeat:auto,no-repeat}@media screen and (min-width:775px){.bg.svelte-1pnr8ds{background-position:right,left;background-attachment:fixed,fixed;background-repeat:auto,repeat-y;background-size:cover,120px}}a{font-weight:700}nav.svelte-2cyc0n{z-index:20;position:fixed;top:0;left:0;right:0;background-color:#fff;font-weight:300;padding:0 1em 1px;transition:box-shadow .2s;margin:0;display:flex;flex-flow:row nowrap}nav.elevated.svelte-2cyc0n{box-shadow:0 0 .5rem rgba(0,0,0,.3),0 1px #eee}.brand-logo.svelte-2cyc0n{height:1em;transform:scale(1) translateX(0);transition:transform .6s ease-in-out}ul.svelte-2cyc0n{margin:0;padding:0;flex-shrink:0}ul.svelte-2cyc0n:first-child{flex-grow:1}ul.svelte-2cyc0n::after{content:'';display:block;clear:both}li.svelte-2cyc0n{display:block;float:left}.selected.svelte-2cyc0n{position:relative;display:inline-block;font-weight:700}.selected.svelte-2cyc0n::after{position:absolute;content:'';width:calc(100% - 1em);height:2px;background-color:#0460a9;display:block;bottom:-1px}a.svelte-2cyc0n{text-decoration:none;padding:1em .5em;display:block}.summary.svelte-dzuyvs{margin:0 0 1em;padding:1em;border-top:3px solid #000;transition:transform .2s;background-color:#fff;box-shadow:-1em 0 #fff}@media screen and (min-width:775px){.summary.svelte-dzuyvs{float:right;width:20rem;transform:translateX(1rem)}}@media screen and (min-width:1200px){.summary.svelte-dzuyvs{float:right;width:20rem;transform:translateX(3rem)}}.content.svelte-dzuyvs h1,h2,h3,h4,h5,h6{margin:0 0 .5em 0;line-height:1.2}.content.svelte-dzuyvs h1,h1.svelte-dzuyvs{font-weight:700;font-size:3rem;margin-bottom:3rem}.content.svelte-dzuyvs h2{font-size:2rem;font-weight:400}.content.svelte-dzuyvs :not(h1):not(hr)+h2{border-top:1px solid #000;padding-top:1em;margin-top:1em}.content.svelte-dzuyvs h3{color:#0460a9}.content.svelte-dzuyvs pre{background-color:#f9f9f9;box-shadow:inset 1px 1px 5px rgba(0,0,0,.05);padding:.5em;border-radius:2px;overflow-x:auto}.content.svelte-dzuyvs pre code{background-color:transparent;padding:0}.content.svelte-dzuyvs ul{line-height:1.5}.content.svelte-dzuyvs li{margin:0 0 .5em 0}.content.svelte-dzuyvs img{max-width:100%}.content.svelte-dzuyvs table{margin:1em 2em;border-collapse:collapse}.content.svelte-dzuyvs th{text-align:left;padding:0 .5em}.content.svelte-dzuyvs thead th{color:#0460a9;font-weight:700;text-transform:uppercase;border-bottom:1px solid #0460a9}.content.svelte-dzuyvs td{text-align:left;padding:.25em .5em}.callout-wrapper.svelte-dzuyvs>div{margin-top:4rem;margin-left:-1rem;margin-right:-1rem}@media screen and (min-width:775px){.callout-wrapper.svelte-dzuyvs>div{margin-left:calc(-75px - 3rem);margin-right:-4rem}}.btn.svelte-zemmdc,a[role=button]{border:1px solid #000;background-color:#fff;color:#000;transition:color .2s,background-color .2s;font-weight:700;display:inline-flex;align-items:center;justify-content:center;text-decoration:none;padding:.25em .75em;white-space:nowrap;cursor:pointer}.btn.svelte-zemmdc:hover,a[role=button]:hover{background-color:#000;color:#fff}.btn.primary.svelte-zemmdc{border-color:#fff;background-color:#0460a9;color:#fff}.btn.primary.svelte-zemmdc:hover{background-color:#fff;color:#0460a9}.callout.svelte-bf2rc0{background-color:#0460a9;color:#fff;padding:2rem 1rem}@media screen and (min-width:775px){.callout.svelte-bf2rc0{padding-left:calc(75px + 3rem)}}.callout-content.svelte-bf2rc0{max-width:50rem}footer.svelte-1o8yfks{display:flex;flex-flow:row nowrap;font-size:.75rem;padding:.5rem .75rem;align-items:center;border-top:1px solid #eee;position:fixed;bottom:0;left:0;right:0;background-color:#fff}@media screen and (max-width:499px){footer.svelte-1o8yfks{text-align:center;flex-flow:column nowrap}}.footer-left.svelte-1o8yfks{flex-grow:1}.footer-right.svelte-1o8yfks{flex-grow:0}a.svelte-1o8yfks{color:rgba(0,0,0,.5);text-decoration:none;font-weight:400}a.svelte-1o8yfks:hover{text-decoration:underline}.social.svelte-1o8yfks{font-size:.825rem}.social.svelte-1o8yfks a.svelte-1o8yfks{border:1px solid #000;background-color:#fff;color:#000;transition:color .2s,background-color .2s;width:1.5rem;height:1.5rem;display:inline-flex;align-items:center;justify-content:center;text-decoration:none;margin:0 .125rem}.social.svelte-1o8yfks a.svelte-1o8yfks:hover{background-color:#000;color:#fff}.sep.svelte-1o8yfks::before{content:'|'}</style> <noscript id=sapper-head-start></noscript><link href=/favicon.ico rel=icon type=image/ico><link href=/manifest.json rel=manifest><title>Yet Another Pipeline</title><noscript id=sapper-head-end></noscript> <link href=/client/1ef9048e76cf74561a9e/main.js rel=preload as=script><link href=/client/1ef9048e76cf74561a9e/projects_$slug.4.js rel=preload as=script></head> <body> <div id=sapper> <nav class=svelte-2cyc0n> <ul class=svelte-2cyc0n> <li class=svelte-2cyc0n> <a href="" class=svelte-2cyc0n>NIBR Open Source</a> </li> </ul> <ul class=svelte-2cyc0n> <li class=svelte-2cyc0n> <a href=https://www.novartis.com class=svelte-2cyc0n rel="nofollow noreferrer" target=_blank> <img class="svelte-2cyc0n brand-logo" src=/novartis-logo.png alt="Novartis logo"> </a> </li> </ul> </nav> <div class="contentPage svelte-1pnr8ds bg" style="background-image:linear-gradient(to right,rgba(255,255,255,0),rgba(255,255,255,0) 75px,rgba(255,255,255,255) 75px,rgba(255,255,255,255)),url(/blue-carbon.png)"> <main class="contentPage svelte-1pnr8ds"> <h1 class=svelte-dzuyvs>Yet Another Pipeline</h1> <div class="svelte-dzuyvs summary"> <h2>Project details</h2> <p>Released 11/6/2014</p> <p><p>YAP is an extensible parallel framework, written in Python using <a href=http://www.open-mpi.org/ >OpenMPI</a> libraries. It allows researchers to quickly build high throughput big data pipelines without extensive knowledge of parallel programming. The user interacts with the framework through simple configuration files to capture analysis parameters and user directed metadata, enabling reproducible research. Using YAP, analysts have been able to achieve a significant speed up of up to 36× in <a href=https://en.wikipedia.org/wiki/RNA-Seq>RNASeq</a> workflow execution time.</p> <p>YAP has been designed to be scalable and flexible. We have implemented YAP with a focus on next-generation sequencing (NGS), to meet the large data processing challenges at NIBR. However, the framework can be easily adapted for any kind of analysis. It can be executed on your local Linux workstations or large HPC cluster systems. The framework achieves efficiency by implementing optimal data handling mechanisms such as, parallel data distribution, avoiding file I/O using data streams and named pipes.</p> <p></p> <a href=https://github.com/novartis/yap class=svelte-zemmdc class=btn rel="nofollow noreferrer" target=_blank component=a> GitHub repo </a> </div> <div class="svelte-dzuyvs content"> <h2> YAP compared to analysts' scripts </h2> <table> <thead> <tr> <th>Analysis</th> <th>Data size</th> <th>Number of cores</th> <th>Analyst methods (hrs)</th> <th>YAP (hrs)</th> <th>Speed-up</th> </tr> </thead> <tbody> <tr> <th>RNASeq QC and Counts</th> <td>3 billion reads (150 samples)</td> <td>500</td> <td>325.6</td> <td>9</td> <td> 36× </td> </tr> <tr> <th>Bacterial studies using Mothur</th> <td>230,000 reads</td> <td>72</td> <td>90</td> <td>12</td> <td>8×</td> </tr> <tr> <th>ChIPSeq Peak Calls</th> <td>190 million reads (6 samples)</td> <td>12</td> <td>9.3</td> <td>4.5</td> <td>2×</td> </tr> <tr> <th>EQP</th> <td>400 million reads (5 samples)</td> <td>60</td> <td>45</td> <td>12</td> <td>4×</td> </tr> </tbody> </table> <div class=container style=text-align:center> <div class=row> <div class=col-sm-3></div> <div class=col-sm-6> <table class="table table-striped table-bordered"> <thead> <tr> <th></th> <th>Traditional method</th> <th>YAP</th> </tr> </thead> <tbody> <tr> <th>I/O steps</th> <td>1400 file reads<br>1200 file writes</td> <td>200 file reads<br>800 file writes</td> </tr> <tr> <th>Jobs spawned</th> <td>1500</td> <td>1 MPI job</td> </tr> <tr> <td colspan=3> File-based reads reduced by <span class="label label-success">70%</span> <br> File-based writes reduced by <span class="label label-success">30%</span> </td> </tr> </tbody> </table> </div> </div> </div> <h2> Example Analysis Output </h2> <p> The following images are the results of various applications run within the YAP framework, such as FastQC, FastQScreen, PicardTools, etc. </p> <img class=scale-image src=/projects/yap/images/sample_output_images.png> <p> <strong>YAP consolidates results from across the samples for the various packages, such as gene counts from HTSeq and normalized counts from Cufflinks.</strong> </p> <div class=row> <div class=col-md-6> <h5>HTSeq gene counts</h5> <table class="table table-striped table-bordered"> <thead> <tr> <th> SAMPLE </th> <th> CR560274_1 </th> <th> CR560457_1 </th> <th> CR560502_1 </th> <th> CR560562_1 </th> <th>...</th> </tr> </thead> <tbody> <tr> <td> NM_000014 </td> <td> 34 </td> <td> 13 </td> <td> 35 </td> <td> 34 </td> <td></td> </tr> <tr> <td> NM_000015 </td> <td> 2 </td> <td> 1 </td> <td> 1 </td> <td> 1 </td> <td></td> </tr> <tr> <td> NM_000016 </td> <td> 0 </td> <td> 0 </td> <td> 0 </td> <td> 0 </td> <td></td> </tr> <tr> <td> NM_000017 </td> <td> 27 </td> <td> 11 </td> <td> 9 </td> <td> 3 <td></td> </tr> <tr><td>NM_000018<td>0<td>0<td>0<td>0<td></tr> <tr><td>NM_000019<td>18<td>48<td>17<td>14<td></tr> <tr><td>NM_000020<td>0<td>0<td>0<td>0<td></tr> <tr><td>NM_000021<td>0<td>0<td>0<td>0<td></tr> <tr> <td>⋮<td><td><td><td><td>⋱</td> </tr> </tbody> </table> </div> <div class=col-md-6> <h5> Normalized counts from Cufflinks </h5> <table class="table table-striped table-bordered"> <tbody> <tr style=font-weight:700> <td> SAMPLE </td> <td colspan=2> CR560274_1 </td> <td colspan=2> CR560457_1 </td> <td colspan=2> CR560502_1 </td> <td>...</tr> <tr style=font-weight:700> <td> TRACKING_ID </td> <td> FPKM </td> <td> FPKM_Status </td> <td> FPKM </td> <td> FPKM_Status </td> <td> FPKM </td> <td class=td10> FPKM_Status </td> <td></tr> <tr> <td> NM_000014<br>|chr12:9220303-9268558| </td> <td> 0.187039 </td> <td> OK </td> <td> 0 </td> <td> OK </td> <td> 0.134608 </td> <td class=td10> OK </td> <td></tr> <tr> <td> NM_000015<br>|chr8:18248754-18258723| </td> <td> 0.218917 </td> <td> OK </td> <td> 0.152739 </td> <td> OK </td> <td> 0.13776 </td> <td class=td10> OK </td> <td></tr> <tr> <td> NM_000016<br>|chr1:76190042-76229355| </td> <td> 2.02618 </td> <td> OK </td> <td> 8.25528 </td> <td> OK </td> <td> 1.7346 </td> <td class=td10> OK </td> <td></tr> <tr><td> NM_000017<br>|chr12:121163570-121177811|<td>1.40608<td>OK<td>0.980779<td>OK<td>0.708088<td>OK <td></tr> <tr> <td>⋮<td><td><td><td><td><td><td>⋱</td> </tr> </tbody> </table> </div> </div> <div class=container-fluid><div class=row><div class=col-xs-4> <h2> Reproducible research </h2> Here's an example of the metadata automatically collected during a YAP run. By storing the commands and parameters used to run the job, YAP allows scientists to reproduce their analysis results at later points. </div> <div class=col-xs-8 style=max-height:600px;overflow-y:scroll> <code><pre style=background-color:#111;color:#aaa>
------------------------------ YAP ANALYSIS SUMMARY FOR WORKFLOW = yap2.3_test ------------------------------
<span style=color:#fff>Operating System Information= Linux yourhostname 2.6.18-371.9.1.el5#1 SMP Tue May 13 06:52:49 EDT 2014x86_64
USER= my_user
YAP SOURCE= /YAP/opensource/
Python Source= 2.7.5 (default, Sep 10 2013, 17:21:36)  [GCC 4.1.2 20080704 (Red Hat 4.1.2-54)]
Analysis Start Time For Workflow : yap2.3_test 2014/09/04 15:55:44
YAP analysis general metadata:
1.comment:120 chars
2.analyst_name:120 chars
3.organisation_name:NIBR
Instrument Type= Illumina
Specimen Information= [tissue type]
Workflow type= rnaseq
Number of input files= 2
Number of processors= 6
Input files path for the workflow= /examples/sample_input
Input file provided:
1.RN0000108D_1 => /examples/sample_input/RN0000108D_1.fq
		  /examples/sample_input/RN0000108D_2.fq

Output file path for the workflow= /test_output/yap2.3_test
Sequence data type= paired end
Input file format= fastq
Maximum read length= 150
File chunk size (in megabytes)= 1024
Data distribution method=chunk_based
Output file path= /test_output/
-------------------------------------------------------------------------------------------------------------
Analysis stages :
Preprocess analysis= yes
Reference Sequence Alignment=yes
Postprocess Analysis= yes
-------------------------------------------------------------------------------------------------------------
Preprocess Analysis commands:
Barcodes information:
no_barcode_specified :
1. command name= fastq_screen,command line= /packages/FastQScreen/v0.4.1/fastq_screen --subset 500000 --paired --outdir output_directory --conf fastq_screen_v0.4.1.conf --aligner bowtie
2. command name= fastqc,command line= /packages/fastqc/0.10.1/fastqc --outdir output_directory --extract --threads 12
-------------------------------------------------------------------------------------------------------------
Aligner commands:
1. command name= bowtie,command line= /packages/bowtie/1.0.0/bowtie  /accessory_files/indexes/bowtie/hg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1 pipe1 -2 pipe2  >output_file.sam
Alignment output data sort order= both
-------------------------------------------------------------------------------------------------------------
Samples re-grouped in this workflow:
None.
-------------------------------------------------------------------------------------------------------------
Potprocess analysis commands:
1. command type= :begin
	command input : ['input_file_type *junctions.bed*', 'input_directory aligner_output']
	1. command name= yap_junction_count,command line= yap_junction_count -exon_coordinates_file /accessory_files/human-ucsc-final_exon_coord.bed -exon_CoordToNumber_file /accessory_files/human-ucsc-final_exon_coord_number.bed -i - -o output_file
2. command type= :begin
	command input : ['input_file_type *queryname*.sam', 'input_directory aligner_output']
	1. command name= htseq-count,command line= /packages/python/2.6.5_gnu/bin/htseq-count -s no -q  file_based_input  /accessory_files/human-ucsc-refGene.gtf  >output_file.out
3. command type= :begin_tee
	command input : ['input_directory aligner_output', 'input_file_type *coordinate*']
	1. command name= yap_exon_count,command line= yap_exon_count -f 1.0 -exon_coordinates_file /accessory_files/human-ucsc-final_exon_coord.bed -exon_CoordToNumber_file /accessory_files/human-ucsc-final_exon_coord_number.bed -i - -o output_file
	2. command name= CollectAlignmentSummaryMetrics,command line= java -Xmx1g -jar /packages/picard-tools/1.89/CollectAlignmentSummaryMetrics.jar VALIDATION_STRINGENCY= SILENT I= /dev/stdin O= output_file.txt IS_BISULFITE_SEQUENCED= true ASSUME_SORTED= True REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa QUIET= True
	3. command name= QualityScoreDistribution,command line= java -Xmx1g -jar /packages/picard-tools/1.89/QualityScoreDistribution.jar VALIDATION_STRINGENCY= SILENT I= /dev/stdin O= output_file.txt ASSUME_SORTED= true REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa CHART= output_file.pdf ALIGNED_READS_ONLY= true
	4. command name= MeanQualityByCycle,command line= java -Xmx1g -jar /packages/picard-tools/1.89/MeanQualityByCycle.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa I= /dev/stdin O= output_file.txt CHART= output_file.pdf ALIGNED_READS_ONLY= true
	5. command name= CollectGcBiasMetrics,command line= java -Xmx1g -jar /packages/picard-tools/1.89/CollectGcBiasMetrics.jar VALIDATION_STRINGENCY= SILENT I= /dev/stdin O= output_file.txt SUMMARY_OUTPUT= output_file_summary.txt CHART= output_file.pdf ASSUME_SORTED= true REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa
	6. command name= CollectRnaSeqMetrics,command line= java -Xmx1g -jar  /packages/picard-tools/1.89/CollectRnaSeqMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REF_FLAT= /db/yap/ucsc/may_02_2013/gtf/hg19/human_refflat_for_picard.gff RIBOSOMAL_INTERVALS= /db/yap/ucsc/may_02_2013/gtf/hg19/Homo_sapiens_assembly19.rRNA.interval_list STRAND_SPECIFICITY= NONE I= /dev/stdin O= output_file.txt CHART_OUTPUT= output_file.pdf REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa
	7. command name= CalculateHsMetrics,command line= /packages/picard-tools/1.89/CalculateHsMetrics.jar VALIDATION_STRINGENCY= SILENT BAIT_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed INPUT= /dev/stdin OUTPUT= output_file.txt REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa
	8. command name= CollectTargetedPcrMetrics,command line= java -Xmx1g -jar /packages/picard-tools/1.89/CollectTargetedPcrMetrics.jar VALIDATION_STRINGENCY= SILENT AMPLICON_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed INPUT= /dev/stdin OUTPUT= output_file.txt REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa
4. command type= :begin
	command input : ['input_file_type *coordinate*', 'input_directory aligner_output']
	1. command name= CollectInsertSizeMetrics,command line= java -Xmx1g -jar  /packages/picard-tools/1.89/CollectInsertSizeMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true I= file_based_input O= output_file.txt H= output_file.pdf TMP_DIR= /scratch/$USER HISTOGRAM_WIDTH= 500
5. command type= :begin
	command input : ['input_file_type *coordinate*', 'input_directory aligner_output']
	1. command name= MarkDuplicates,command line= java -Xmx1g -jar  /packages/picard-tools/1.89/MarkDuplicates.jar VALIDATION_STRINGENCY= SILENT TMP_DIR= /scratch/$USER MAX_RECORDS_IN_RAM= 1000000 MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP= 2000 ASSUME_SORTED= true I= file_based_input O= output_file.bam METRICS_FILE= output_file.txt
6. command type= :begin
	command input : ['input_file_type *coordinate*', 'input_directory aligner_output']
	1. command name= cufflinks,command line= /packages/cufflinks/2.1.1/cufflinks  file_based_input -o output_directory -p 12 -G /accessory_files/human-ucsc-refGene.gtf
-------------------------------------------------------------------------------------------------------------

******************* YAP CHECK SUMMARY *******************
* --Syntax check          : Passed                      *
* --Compatibility check   : Passed                      *
* --File paths check      : Passed With Warnings        *
*********************************************************
* YAP Configuration overall check status: Passed With Warnings</span>

--------------------------------------- YAP Check Error/Warning Info ---------------------------------------
-------------------------------------------------------------------------------------------------------------
--YAP Configuration File paths check status: Passed With Warnings
Warning: At Line: 21 in file: bowtie_1.0.0_configuration.cfg. Files were found using basename in /db/nibrgenome/NG00006.0/indexes/bowtie/hg19. Please make sure that command: bowtie can work with basenames.


-------------------------------------------------------------------------------------------------------------
-------------------------- YAP configurations check end for Workflow = yap2.3_test --------------------------
-------------------- PROVENANCE --------------------

PREPROCESS:

	/packages/FastQScreen/v0.4.1/fastq_screen --subset 500000 --paired --outdir /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/preprocess_output --conf fastq_screen_v0.4.1.conf --aligner bowtie /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq

	/packages/fastqc/0.10.1/fastqc --outdir /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/preprocess_output --extract --threads 12 /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq

ALIGNMENT :

	INPUT: /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq and /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq chunk number= 0

	/packages/bowtie/1.0.0/bowtie  /db/nibrgenome/NG00006.0/indexes/bowtie/hg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000003008281_0.797636572311_pipe_0_0_1  -2  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000003008281_0.797636572311_pipe_0_0_2   >/test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000000.sam

	samtools view -bhS - samtools sort -on -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000000_queryname | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000000_queryname.sam

	samtools view -bhS - samtools sort -o -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000000_coordinate | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000000_coordinate.sam

	INPUT: /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq and /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq chunk number= 1

	/packages/bowtie/1.0.0/bowtie  /db/nibrgenome/NG00006.0/indexes/bowtie/hg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000013008281_0.797636572311_pipe_1_0_1  -2  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000013008281_0.797636572311_pipe_1_0_2   >/test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000001.sam

	samtools view -bhS - samtools sort -on -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000001_queryname | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000001_queryname.sam

	samtools view -bhS - samtools sort -o -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000001_coordinate | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000001_coordinate.sam

	INPUT: /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq and /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq chunk number= 2

	/packages/bowtie/1.0.0/bowtie  /db/nibrgenome/NG00006.0/indexes/bowtie/hg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000023008281_0.797636572311_pipe_2_0_1  -2  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000023008281_0.797636572311_pipe_2_0_2   >/test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000002.sam

	samtools view -bhS - samtools sort -on -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000002_queryname | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000002_queryname.sam

	samtools view -bhS - samtools sort -o -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000002_coordinate | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000002_coordinate.sam

	INPUT: /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq and /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq chunk number= 3

	/packages/bowtie/1.0.0/bowtie  /db/nibrgenome/NG00006.0/indexes/bowtie/hg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000033008281_0.797636572311_pipe_3_0_1  -2  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000033008281_0.797636572311_pipe_3_0_2   >/test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000003.sam

	samtools view -bhS - samtools sort -on -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000003_queryname | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000003_queryname.sam

	samtools view -bhS - samtools sort -o -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000003_coordinate | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000003_coordinate.sam

	INPUT: /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq and /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq chunk number= 4

	/packages/bowtie/1.0.0/bowtie  /db/nibrgenome/NG00006.0/indexes/bowtie/hg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000043008281_0.797636572311_pipe_4_0_1  -2  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000043008281_0.797636572311_pipe_4_0_2   >/test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000004.sam

	samtools view -bhS - samtools sort -on -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000004_queryname | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000004_queryname.sam

	samtools view -bhS - samtools sort -o -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000004_coordinate | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000004_coordinate.sam

	INPUT: /db/yap/benchmark/robin_50_samples/RN0000108D_1.fq and /db/yap/benchmark/robin_50_samples/RN0000108D_2.fq chunk number= 5

	/packages/bowtie/1.0.0/bowtie  /db/nibrgenome/NG00006.0/indexes/bowtie/hg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000053008281_0.797636572311_pipe_5_0_1  -2  /scratch/kulkatr1//kulkatr1/yap_temp/aligner_RN0000108D_1_0000053008281_0.797636572311_pipe_5_0_2   >/test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000005.sam

	samtools view -bhS - samtools sort -on -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000005_queryname | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000005_queryname.sam

	samtools view -bhS - samtools sort -o -m 100000000 - /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000005_coordinate | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000005_coordinate.sam

MERGE ALIGNMENT OUTPUT :

	samtools merge -n  -  /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000002_queryname.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000003_queryname.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000005_queryname.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000000_queryname.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000001_queryname.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000004_queryname.bam  | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_queryname.sam

	samtools merge  -  /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000005_coordinate.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000001_coordinate.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000000_coordinate.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000002_coordinate.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000003_coordinate.bam /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/aligner_RN0000108D_1_000004_coordinate.bam  | samtools view -h - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam

POSTPROCESS :

	INPUT: /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_queryname.sam

	/packages/python/2.6.5_gnu/bin/htseq-count -s no -q  /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_queryname.sam  /accessory_files/human-ucsc-refGene.gtf  >/test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_htseq-count.out

	INPUT: /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam

	yap_exon_count -f 1.0 -exon_coordinates_file /accessory_files/human-ucsc-final_exon_coord.bed -exon_CoordToNumber_file /accessory_files/human-ucsc-final_exon_coord_number.bed -i - -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_yap_exon_count

	java -Xmx1g -jar /packages/picard-tools/1.89/CollectAlignmentSummaryMetrics.jar VALIDATION_STRINGENCY= SILENT I= /dev/stdin O= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectAlignmentSummaryMetrics.txt IS_BISULFITE_SEQUENCED= true ASSUME_SORTED= True REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa QUIET= True

	java -Xmx1g -jar /packages/picard-tools/1.89/QualityScoreDistribution.jar VALIDATION_STRINGENCY= SILENT I= /dev/stdin O= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_QualityScoreDistribution.txt ASSUME_SORTED= true REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa CHART= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_QualityScoreDistribution.pdf ALIGNED_READS_ONLY= true

	java -Xmx1g -jar /packages/picard-tools/1.89/MeanQualityByCycle.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa I= /dev/stdin O= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_MeanQualityByCycle.txt CHART= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_MeanQualityByCycle.pdf ALIGNED_READS_ONLY= true

	java -Xmx1g -jar /packages/picard-tools/1.89/CollectGcBiasMetrics.jar VALIDATION_STRINGENCY= SILENT I= /dev/stdin O= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectGcBiasMetrics.txt SUMMARY_OUTPUT= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectGcBiasMetrics_summary.txt CHART= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectGcBiasMetrics.pdf ASSUME_SORTED= true REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa

	java -Xmx1g -jar  /packages/picard-tools/1.89/CollectRnaSeqMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REF_FLAT= /db/yap/ucsc/may_02_2013/gtf/hg19/human_refflat_for_picard.gff RIBOSOMAL_INTERVALS= /db/yap/ucsc/may_02_2013/gtf/hg19/Homo_sapiens_assembly19.rRNA.interval_list STRAND_SPECIFICITY= NONE I= /dev/stdin O= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectRnaSeqMetrics.txt CHART_OUTPUT= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectRnaSeqMetrics.pdf REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa

	/packages/picard-tools/1.89/CalculateHsMetrics.jar VALIDATION_STRINGENCY= SILENT BAIT_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed INPUT= /dev/stdin OUTPUT= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CalculateHsMetrics.txt REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa

	java -Xmx1g -jar /packages/picard-tools/1.89/CollectTargetedPcrMetrics.jar VALIDATION_STRINGENCY= SILENT AMPLICON_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= /accessory_files/TruSeq_exome_targeted_regions_for_picard.bed INPUT= /dev/stdin OUTPUT= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectTargetedPcrMetrics.txt REFERENCE_SEQUENCE= /db/nibrgenome/NG00006.0/fasta/hg19.fa

	INPUT: /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam

	java -Xmx1g -jar  /packages/picard-tools/1.89/CollectInsertSizeMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true I= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam O= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectInsertSizeMetrics.txt H= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_CollectInsertSizeMetrics.pdf TMP_DIR= /scratch/$USER HISTOGRAM_WIDTH= 500

	INPUT: /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam

	java -Xmx1g -jar  /packages/picard-tools/1.89/MarkDuplicates.jar VALIDATION_STRINGENCY= SILENT TMP_DIR= /scratch/$USER MAX_RECORDS_IN_RAM= 1000000 MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP= 2000 ASSUME_SORTED= true I= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam O= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_MarkDuplicates.bam METRICS_FILE= /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output/RN0000108D_1_MarkDuplicates.txt

	INPUT: /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam

	/packages/cufflinks/2.1.1/cufflinks  /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/aligner_output/RN0000108D_1_coordinate.sam -o /test_output/yap2.3_test/RN0000108D_1/no_barcode_specified/postprocess_output -p 12 -G /accessory_files/human-ucsc-refGene.gtf

--------------------Analysis End Time For Workflow : yap2.3_test 2014/09/04 19:43:14--------------------</pre> </code> </div> </div> </div> <h2> <a href=#requirements class=anchor aria-hidden=true name=user-content-requirements><span class="octicon octicon-link"></span></a>Requirements</h2> <p><strong>YAP only runs on Linux systems!</strong></p> <h3>Dependencies: </h3> <p>The following dependencies have to be first installed in your environment. Once installed, make sure these dependencies are added to your path.</p> <ul class=task-list> <li>Recent versions of gcc (gcc 4.8.x is well tested)</li> <li>Python 2.7.7</li> <li>Openmpi 1.6.5</li> <li>Python modules: <ul class=task-list> <li>MPI4py - 1.3</li> <li>PyPdf - 1.13</li> <li>Numpy - 1.7.1</li> <li>netsa-utils - 1.4.3</li> </ul> </li> <li>bedtools - 2.15.0</li> <li>samtools - 0.1.18</li> </ul> <h3>System Configuration:</h3> <p> YAP provides a framework to run external tools and data, so the tools used in the workflows drive the system requirements. It can be installed on multicore linux workstation with a decent amount of memory for small data, or on large cluster systems to scale optimally for large data processing. The framework has been tested extensively for <abbr title="Next-Generation Sequencing">NGS</abbr> data on clusters with minimum system configuration of 8-12 cores and 24-48 GB memory. </p> <h2>YAP Setup</h2> <ul class=task-list> <li>Download the yap source from <a href=https://github.com/Novartis/yap/archive/master.zip>here</a> </li> <li> <p>Uncompress the source directory </p> <p>for example: uncompress the directory as <code>/home/packages/YAP</code></p> </li> <li> <p>Set <code>YAP_HOME</code> environment variable to the source directory.</p> <p><pre><code>$ export YAP_HOME=/home/packages/YAP</code></pre></p> </li> <li> <p>Add bin directory to path</p> <pre><code>$ export PATH=$PATH:$YAP_HOME/bin/</code></pre> </li> <li> <p>Set <code>YAP_LOCAL_TEMPDIR</code> environment variable for temporary computation. For optimum performance point this directory to a location which is local to the machine. </p> <pre><code>$ export YAP_LOCAL_TEMPDIR=/scratch/username/yap_temp</code></pre> </li> </ul><p><strong>Verification</strong></p> <pre><code>$ echo $YAP_HOME
    /home/packages/YAP

$ echo $YAP_LOCAL_TEMPDIR
    /scratch/username/yap_temp

$ which yap
    /home/packages/YAP/bin/yap
</code></pre> <h1> <a href=#running-a-yap-job class=anchor aria-hidden=true name=user-content-running-a-yap-job><span class="octicon octicon-link"></span></a>Running a YAP job</h1> <p>Once you've set your environment, it is best to run a quick demo job to get the feel of running YAP. The following section is meant to be interactive and hence you would need Linux account access and access to the cluster.</p> <p>After downloading the project, please see the demo configuration files in <code>yap/cfg</code>.</p> <p>There are 3 stages in YAP - Preprocess, Alignment and Postprocess. You can have command level control of these three stages in the namesake configuration files and a workflow level control in the <code>workflow_configuration.</code></p> <table class="table table-striped"> <thead><tr> <th>Configuration</th> <th>Purpose</th> </thead> <tbody> <tr> <td>aligner_configuration</td> <td>bwa, bowtie, bowtie2, tophat or insert your own aligner</td> </tr> <tr> <td>postprocess_configuration</td> <td>postalignment packages, generate counts or metrics</td> </tr> <tr> <td>preprocess_configuration</td> <td>pre-alignment packages to massage your seqdata</td> </tr> <tr> <td>workflow_configuration</td> <td>manage metadata, specify input files, paths and output directories</td> </tr> <tr> <td>yap_sge</td> <td>submitting your job to the cluster</td> </tr> </tbody> </table><p>The demo runs a RNASeq QC and counts workflow consisting of:</p> <ul class=task-list> <li>Preprocess: FastQC, Fastqscreen</li> <li>Alignment: Bowtie, both queryname and coordinate sorted</li> <li>Postprocess: yap junction and exon counts, Picard tools (PostQC), HTSeq (Raw counts) and Cufflinks (normalized counts)</li> </ul><p>We run this workflow on 2 nodes on the UGE cluster.</p> <p>To run the yap_demo job, we next need to check to see if our configuration files are correct using the command.</p> <pre><code>cd &lt;your_working directory>
yap --check workflow_configuration.cfg
</code></pre> <p>The <code>yap --check</code> command checks to see</p> <ul class=task-list> <li>If all paths specified are valid</li> <li>If YAP finds the appropriate input files</li> <li>Checks for syntax errors</li> <li>Lists commands to be executed.</li> <li>Gives section-wise error/warning report.</li> </ul> <h2>Running the YAP job</h2> <pre><code>mpirun -n &lt;number_of_cores> yap workflow_configuration.cfg
</code></pre> <p>If you have a SGE environment, pass the number of slots into the <code>$NSLOTS</code> variable.</p> <script> $('abbr').tooltip(); </script> </div> <div class="svelte-dzuyvs callout-wrapper"> <div class="svelte-bf2rc0 callout"> <div class="svelte-bf2rc0 callout-content"> <h2>Interested in NIBR Engineering?</h2> <p>At NIBR, you'll be at the forefront of technology — helping to shape it, develop it, and make it impactful. Partnering with scientists, our engineers create cutting-edge, state-of-the-art solutions that accelerate drug discovery and ultimately improve patients’ lives.</p> <a href=https://www.novartis.com/careers/careers-research class="primary svelte-zemmdc" class="btn primary" rel="nofollow noopener" target=_blank variant=primary> Learn more </a> </div> </div> </div> </main> </div> <footer class=svelte-1o8yfks> <div class="svelte-1o8yfks footer-left"> © 2019 <a href=https://www.novartis.com class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank>Novartis AG</a> <span class="svelte-1o8yfks sep"></span> <a href=https://www.novartis.com/terms-use class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank> Terms of use </a> <span class="svelte-1o8yfks sep"></span> <a href=https://www.novartis.com/privacy-policy class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank> Privacy policy </a> <span class="svelte-1o8yfks sep"></span> <a href=mailto:open.source@novartis.com class=svelte-1o8yfks> open.source@novartis.com </a> </div> <div class="svelte-1o8yfks footer-right social"> <a href=https://www.github.com/Novartis class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=GitHub> <i class="fab fa-github"></i> </a> <a href=https://www.facebook.com/novartis class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=Facebook> <i class="fab fa-facebook-f"></i> </a> <a href=https://twitter.com/NovartisScience class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=Twitter> <i class="fab fa-twitter"></i> </a> <a href=https://www.linkedin.com/company/novartis-institutes-for-biomedical-research class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=LinkedIn> <i class="fab fa-linkedin-in"></i> </a> </div> </footer></div> <script>__SAPPER__={baseUrl:"",preloaded:[void 0,null,(function(a,b){return {project:{slug:a,metadata:{name:b,abbreviation:"YAP",projectName:a,launched:"2014-11-06T00:00:00.000Z",icon:"yap_logo.png",summary:"YAP allows researchers to quickly build high throughput big data pipelines without extensive\nknowledge of parallel programming.\n",description:"\u003Cp\u003EYAP is an extensible parallel framework, written in Python using \u003Ca href=\"http:\u002F\u002Fwww.open-mpi.org\u002F\"\u003EOpenMPI\u003C\u002Fa\u003E libraries. It\nallows researchers to quickly build high throughput big data pipelines without extensive knowledge of parallel\nprogramming. The user interacts with the framework through simple configuration files to capture analysis parameters\nand user directed metadata, enabling reproducible research. Using YAP, analysts have been able to achieve a\nsignificant speed up of up to 36× in \u003Ca href=\"https:\u002F\u002Fen.wikipedia.org\u002Fwiki\u002FRNA-Seq\"\u003ERNASeq\u003C\u002Fa\u003E workflow execution time.\u003C\u002Fp\u003E\n\u003Cp\u003EYAP has been designed to be scalable and flexible. We have implemented YAP with a focus on next-generation sequencing\n(NGS), to meet the large data processing challenges at NIBR. However, the framework can be easily adapted for any kind\nof analysis. It can be executed on your local Linux workstations or large HPC cluster systems. The framework achieves\nefficiency by implementing optimal data handling mechanisms such as, parallel data distribution, avoiding file I\u002FO\nusing data streams and named pipes.\u003C\u002Fp\u003E\n",dateString:"Thu Nov 06 2014",title:b,styles:[],scripts:[]},html:"\n\u003Ch2\u003E YAP compared to analysts' scripts \u003C\u002Fh2\u003E\n\u003Ctable\u003E\n  \u003Cthead\u003E\n    \u003Ctr\u003E\n      \u003Cth\u003EAnalysis\u003C\u002Fth\u003E\n      \u003Cth\u003EData size\u003C\u002Fth\u003E\n      \u003Cth\u003ENumber of cores\u003C\u002Fth\u003E\n      \u003Cth\u003EAnalyst methods (hrs)\u003C\u002Fth\u003E\n      \u003Cth\u003EYAP (hrs)\u003C\u002Fth\u003E\n      \u003Cth\u003ESpeed-up\u003C\u002Fth\u003E\n    \u003C\u002Ftr\u003E\n  \u003C\u002Fthead\u003E\n  \u003Ctbody\u003E\n    \u003Ctr\u003E\n      \u003Cth\u003ERNASeq QC and Counts\u003C\u002Fth\u003E\n      \u003Ctd\u003E3 billion reads (150 samples)\u003C\u002Ftd\u003E\n      \u003Ctd\u003E500\u003C\u002Ftd\u003E\n      \u003Ctd\u003E325.6\u003C\u002Ftd\u003E\n      \u003Ctd\u003E9\u003C\u002Ftd\u003E\n      \u003Ctd\u003E\n        36&times;\n      \u003C\u002Ftd\u003E\n    \u003C\u002Ftr\u003E\n    \u003Ctr\u003E\n      \u003Cth\u003EBacterial studies using Mothur\u003C\u002Fth\u003E\n      \u003Ctd\u003E230,000 reads\u003C\u002Ftd\u003E\n      \u003Ctd\u003E72\u003C\u002Ftd\u003E\n      \u003Ctd\u003E90\u003C\u002Ftd\u003E\n      \u003Ctd\u003E12\u003C\u002Ftd\u003E\n      \u003Ctd\u003E8&times;\u003C\u002Ftd\u003E\n    \u003C\u002Ftr\u003E\n    \u003Ctr\u003E\n      \u003Cth\u003EChIPSeq Peak Calls\u003C\u002Fth\u003E\n      \u003Ctd\u003E190 million reads (6 samples)\u003C\u002Ftd\u003E\n      \u003Ctd\u003E12\u003C\u002Ftd\u003E\n      \u003Ctd\u003E9.3\u003C\u002Ftd\u003E\n      \u003Ctd\u003E4.5\u003C\u002Ftd\u003E\n      \u003Ctd\u003E2&times;\u003C\u002Ftd\u003E\n    \u003C\u002Ftr\u003E\n    \u003Ctr\u003E\n      \u003Cth\u003EEQP\u003C\u002Fth\u003E\n      \u003Ctd\u003E400 million reads (5 samples)\u003C\u002Ftd\u003E\n      \u003Ctd\u003E60\u003C\u002Ftd\u003E\n      \u003Ctd\u003E45\u003C\u002Ftd\u003E\n      \u003Ctd\u003E12\u003C\u002Ftd\u003E\n      \u003Ctd\u003E4&times;\u003C\u002Ftd\u003E\n    \u003C\u002Ftr\u003E\n  \u003C\u002Ftbody\u003E\n\u003C\u002Ftable\u003E\n\n\u003Cdiv class=\"container\" style=\"text-align:center;\"\u003E\n  \u003Cdiv class=\"row\"\u003E\n    \u003Cdiv class=\"col-sm-3\"\u003E\u003C\u002Fdiv\u003E\n    \u003Cdiv class=\"col-sm-6\"\u003E\n      \u003Ctable class=\"table table-striped table-bordered\"\u003E\n        \u003Cthead\u003E\n          \u003Ctr\u003E\n            \u003Cth\u003E\u003C\u002Fth\u003E\n            \u003Cth\u003ETraditional method\u003C\u002Fth\u003E\n            \u003Cth\u003EYAP\u003C\u002Fth\u003E\n          \u003C\u002Ftr\u003E\n        \u003C\u002Fthead\u003E\n        \u003Ctbody\u003E\n          \u003Ctr\u003E\n            \u003Cth\u003EI\u002FO steps\u003C\u002Fth\u003E\n            \u003Ctd\u003E1400 file reads\u003Cbr\u003E1200 file writes\u003C\u002Ftd\u003E\n            \u003Ctd\u003E200 file reads\u003Cbr\u003E800 file writes\u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Cth\u003EJobs spawned\u003C\u002Fth\u003E\n            \u003Ctd\u003E1500\u003C\u002Ftd\u003E\n            \u003Ctd\u003E1 MPI job\u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Ctd colspan=3\u003E\n              File-based reads reduced by \u003Cspan class=\"label label-success\"\u003E70%\u003C\u002Fspan\u003E\n              \u003Cbr\u003E\n              File-based writes reduced by \u003Cspan class=\"label label-success\"\u003E30%\u003C\u002Fspan\u003E\n            \u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n        \u003C\u002Ftbody\u003E\n      \u003C\u002Ftable\u003E\n    \u003C\u002Fdiv\u003E\n  \u003C\u002Fdiv\u003E\n\u003C\u002Fdiv\u003E\n\n\u003Ch2\u003E Example Analysis Output \u003C\u002Fh2\u003E\n\u003Cp\u003E\n\n  The following images are the results of various applications run within the\n  YAP framework, such as FastQC, FastQScreen, PicardTools, etc.\n\n\u003C\u002Fp\u003E\n\u003Cimg src=\"\u002Fprojects\u002Fyap\u002Fimages\u002Fsample_output_images.png\" class=\"scale-image\"\u003E\n\n\u003Cp\u003E\n\n  \u003Cstrong\u003EYAP consolidates results from across the samples for the various\n      packages, such as gene counts from HTSeq and normalized counts from Cufflinks.\u003C\u002Fstrong\u003E\n\n\u003C\u002Fp\u003E\n\n\u003Cdiv class=\"row\"\u003E\n  \u003Cdiv class=\"col-md-6\"\u003E\n    \u003Ch5\u003EHTSeq gene counts\u003C\u002Fh5\u003E\n    \u003Ctable class='table table-striped table-bordered'\u003E\n        \u003Cthead\u003E\n          \u003Ctr\u003E\n            \u003Cth\u003E\n              SAMPLE\n            \u003C\u002Fth\u003E\n            \u003Cth\u003E\n              CR560274_1\n            \u003C\u002Fth\u003E\n            \u003Cth\u003E\n              CR560457_1\n            \u003C\u002Fth\u003E\n            \u003Cth\u003E\n              CR560502_1\n            \u003C\u002Fth\u003E\n            \u003Cth\u003E\n              CR560562_1\n            \u003C\u002Fth\u003E\n            \u003Cth\u003E...\u003C\u002Fth\u003E\n          \u003C\u002Ftr\u003E\n        \u003C\u002Fthead\u003E\n        \u003Ctbody\u003E\n          \u003Ctr\u003E\n            \u003Ctd\u003E\n              NM_000014\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              34\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              13\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              35\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              34\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Ctd\u003E\n              NM_000015\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              2\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              1\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              1\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              1\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Ctd\u003E\n              NM_000016\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Ctd\u003E\n              NM_000017\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              27\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              11\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              9\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              3\n            \u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n\n          \u003Ctr\u003E\u003Ctd\u003ENM_000018\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n          \u003Ctr\u003E\u003Ctd\u003ENM_000019\u003C\u002Ftd\u003E\u003Ctd\u003E18\u003C\u002Ftd\u003E\u003Ctd\u003E48\u003C\u002Ftd\u003E\u003Ctd\u003E17\u003C\u002Ftd\u003E\u003Ctd\u003E14\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n\u003Ctr\u003E\u003Ctd\u003ENM_000020\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n\u003Ctr\u003E\u003Ctd\u003ENM_000021\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E0\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n\n          \u003Ctr\u003E\n            \u003Ctd\u003E&#x22ee;\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E&#x22f1;\u003C\u002Ftd\u003E\n          \u003C\u002Ftr\u003E\n        \u003C\u002Ftbody\u003E\n      \u003C\u002Ftable\u003E\n  \u003C\u002Fdiv\u003E\n  \u003Cdiv class=\"col-md-6\"\u003E\n    \u003Ch5\u003E Normalized counts from Cufflinks \u003C\u002Fh5\u003E\n    \u003Ctable class='table table-striped table-bordered'\u003E\n      \u003Ctbody\u003E\n          \u003Ctr style=\"font-weight: bold;\"\u003E\n            \u003Ctd\u003E\n              SAMPLE\n            \u003C\u002Ftd\u003E\n            \u003Ctd colspan=2\u003E\n              CR560274_1\n            \u003C\u002Ftd\u003E\n            \u003Ctd colspan=2\u003E\n              CR560457_1\n            \u003C\u002Ftd\u003E\n            \u003Ctd colspan=2\u003E\n              CR560502_1\n            \u003C\u002Ftd\u003E\n          \u003Ctd\u003E...\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n          \u003Ctr style=\"font-weight: bold;\"\u003E\n            \u003Ctd\u003E\n              TRACKING_ID\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              FPKM\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              FPKM_Status\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              FPKM\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              FPKM_Status\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              FPKM\n            \u003C\u002Ftd\u003E\n            \u003Ctd class=\"td10\"\u003E\n              FPKM_Status\n            \u003C\u002Ftd\u003E\n          \u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Ctd\u003E\n              NM_000014\u003Cbr\u003E|chr12:9220303-9268558|\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0.187039\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              OK\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              OK\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0.134608\n            \u003C\u002Ftd\u003E\n            \u003Ctd class=\"td10\"\u003E\n              OK\n            \u003C\u002Ftd\u003E\n          \u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Ctd\u003E\n              NM_000015\u003Cbr\u003E|chr8:18248754-18258723|\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0.218917\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              OK\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0.152739\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              OK\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              0.13776\n            \u003C\u002Ftd\u003E\n            \u003Ctd class=\"td10\"\u003E\n              OK\n            \u003C\u002Ftd\u003E\n          \u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n          \u003Ctr\u003E\n            \u003Ctd\u003E\n              NM_000016\u003Cbr\u003E|chr1:76190042-76229355|\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              2.02618\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              OK\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              8.25528\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              OK\n            \u003C\u002Ftd\u003E\n            \u003Ctd\u003E\n              1.7346\n            \u003C\u002Ftd\u003E\n            \u003Ctd class=\"td10\"\u003E\n              OK\n            \u003C\u002Ftd\u003E\n          \u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n          \u003Ctr\u003E\u003Ctd\u003E\n            NM_000017\u003Cbr\u003E|chr12:121163570-121177811|\u003C\u002Ftd\u003E\u003Ctd\u003E1.40608\u003C\u002Ftd\u003E\u003Ctd\u003EOK\u003C\u002Ftd\u003E\u003Ctd\u003E0.980779\u003C\u002Ftd\u003E\u003Ctd\u003EOK\u003C\u002Ftd\u003E\u003Ctd\u003E0.708088\u003C\u002Ftd\u003E\u003Ctd\u003EOK\n            \u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003C\u002Ftr\u003E\n              \u003Ctr\u003E\n              \u003Ctd\u003E⋮\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E\u003C\u002Ftd\u003E\u003Ctd\u003E⋱\u003C\u002Ftd\u003E\n            \u003C\u002Ftr\u003E\n            \u003C\u002Ftbody\u003E\n          \u003C\u002Ftable\u003E\n\n  \u003C\u002Fdiv\u003E\u003C!-- col-md-4 --\u003E\n\u003C\u002Fdiv\u003E\u003C!-- row --\u003E\n\n\u003Cdiv class=\"container-fluid\"\u003E\u003Cdiv class=\"row\"\u003E\u003Cdiv class=\"col-xs-4\"\u003E\n  \u003Ch2\u003E Reproducible research \u003C\u002Fh2\u003E\n\n  Here's an example of the metadata automatically collected during a YAP run. By\n  storing the commands and parameters used to run the job, YAP allows\n  scientists to reproduce their analysis results at later points.\n\n\u003C\u002Fdiv\u003E\n\u003Cdiv class=\"col-xs-8\" style=\"max-height: 600px; overflow-y: scroll;\"\u003E\n  \u003Ccode\u003E\u003Cpre style=\"background-color: #111; color: #aaa;\"\u003E\n------------------------------ YAP ANALYSIS SUMMARY FOR WORKFLOW = yap2.3_test ------------------------------\n\u003Cspan style=\"color:white;\"\u003EOperating System Information= Linux yourhostname 2.6.18-371.9.1.el5#1 SMP Tue May 13 06:52:49 EDT 2014x86_64\nUSER= my_user\nYAP SOURCE= \u002FYAP\u002Fopensource\u002F\nPython Source= 2.7.5 (default, Sep 10 2013, 17:21:36)  [GCC 4.1.2 20080704 (Red Hat 4.1.2-54)]\nAnalysis Start Time For Workflow : yap2.3_test 2014\u002F09\u002F04 15:55:44\nYAP analysis general metadata:\n1.comment:120 chars\n2.analyst_name:120 chars\n3.organisation_name:NIBR\nInstrument Type= Illumina\nSpecimen Information= [tissue type]\nWorkflow type= rnaseq\nNumber of input files= 2\nNumber of processors= 6\nInput files path for the workflow= \u002Fexamples\u002Fsample_input\nInput file provided:\n1.RN0000108D_1 =\u003E \u002Fexamples\u002Fsample_input\u002FRN0000108D_1.fq\n\t\t  \u002Fexamples\u002Fsample_input\u002FRN0000108D_2.fq\n\nOutput file path for the workflow= \u002Ftest_output\u002Fyap2.3_test\nSequence data type= paired end\nInput file format= fastq\nMaximum read length= 150\nFile chunk size (in megabytes)= 1024\nData distribution method=chunk_based\nOutput file path= \u002Ftest_output\u002F\n-------------------------------------------------------------------------------------------------------------\nAnalysis stages :\nPreprocess analysis= yes\nReference Sequence Alignment=yes\nPostprocess Analysis= yes\n-------------------------------------------------------------------------------------------------------------\nPreprocess Analysis commands:\nBarcodes information:\nno_barcode_specified :\n1. command name= fastq_screen,command line= \u002Fpackages\u002FFastQScreen\u002Fv0.4.1\u002Ffastq_screen --subset 500000 --paired --outdir output_directory --conf fastq_screen_v0.4.1.conf --aligner bowtie\n2. command name= fastqc,command line= \u002Fpackages\u002Ffastqc\u002F0.10.1\u002Ffastqc --outdir output_directory --extract --threads 12\n-------------------------------------------------------------------------------------------------------------\nAligner commands:\n1. command name= bowtie,command line= \u002Fpackages\u002Fbowtie\u002F1.0.0\u002Fbowtie  \u002Faccessory_files\u002Findexes\u002Fbowtie\u002Fhg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1 pipe1 -2 pipe2  \u003Eoutput_file.sam\nAlignment output data sort order= both\n-------------------------------------------------------------------------------------------------------------\nSamples re-grouped in this workflow:\nNone.\n-------------------------------------------------------------------------------------------------------------\nPotprocess analysis commands:\n1. command type= :begin\n\tcommand input : ['input_file_type *junctions.bed*', 'input_directory aligner_output']\n\t1. command name= yap_junction_count,command line= yap_junction_count -exon_coordinates_file \u002Faccessory_files\u002Fhuman-ucsc-final_exon_coord.bed -exon_CoordToNumber_file \u002Faccessory_files\u002Fhuman-ucsc-final_exon_coord_number.bed -i - -o output_file\n2. command type= :begin\n\tcommand input : ['input_file_type *queryname*.sam', 'input_directory aligner_output']\n\t1. command name= htseq-count,command line= \u002Fpackages\u002Fpython\u002F2.6.5_gnu\u002Fbin\u002Fhtseq-count -s no -q  file_based_input  \u002Faccessory_files\u002Fhuman-ucsc-refGene.gtf  \u003Eoutput_file.out\n3. command type= :begin_tee\n\tcommand input : ['input_directory aligner_output', 'input_file_type *coordinate*']\n\t1. command name= yap_exon_count,command line= yap_exon_count -f 1.0 -exon_coordinates_file \u002Faccessory_files\u002Fhuman-ucsc-final_exon_coord.bed -exon_CoordToNumber_file \u002Faccessory_files\u002Fhuman-ucsc-final_exon_coord_number.bed -i - -o output_file\n\t2. command name= CollectAlignmentSummaryMetrics,command line= java -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectAlignmentSummaryMetrics.jar VALIDATION_STRINGENCY= SILENT I= \u002Fdev\u002Fstdin O= output_file.txt IS_BISULFITE_SEQUENCED= true ASSUME_SORTED= True REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa QUIET= True\n\t3. command name= QualityScoreDistribution,command line= java -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FQualityScoreDistribution.jar VALIDATION_STRINGENCY= SILENT I= \u002Fdev\u002Fstdin O= output_file.txt ASSUME_SORTED= true REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa CHART= output_file.pdf ALIGNED_READS_ONLY= true\n\t4. command name= MeanQualityByCycle,command line= java -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FMeanQualityByCycle.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa I= \u002Fdev\u002Fstdin O= output_file.txt CHART= output_file.pdf ALIGNED_READS_ONLY= true\n\t5. command name= CollectGcBiasMetrics,command line= java -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectGcBiasMetrics.jar VALIDATION_STRINGENCY= SILENT I= \u002Fdev\u002Fstdin O= output_file.txt SUMMARY_OUTPUT= output_file_summary.txt CHART= output_file.pdf ASSUME_SORTED= true REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n\t6. command name= CollectRnaSeqMetrics,command line= java -Xmx1g -jar  \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectRnaSeqMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REF_FLAT= \u002Fdb\u002Fyap\u002Fucsc\u002Fmay_02_2013\u002Fgtf\u002Fhg19\u002Fhuman_refflat_for_picard.gff RIBOSOMAL_INTERVALS= \u002Fdb\u002Fyap\u002Fucsc\u002Fmay_02_2013\u002Fgtf\u002Fhg19\u002FHomo_sapiens_assembly19.rRNA.interval_list STRAND_SPECIFICITY= NONE I= \u002Fdev\u002Fstdin O= output_file.txt CHART_OUTPUT= output_file.pdf REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n\t7. command name= CalculateHsMetrics,command line= \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCalculateHsMetrics.jar VALIDATION_STRINGENCY= SILENT BAIT_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed INPUT= \u002Fdev\u002Fstdin OUTPUT= output_file.txt REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n\t8. command name= CollectTargetedPcrMetrics,command line= java -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectTargetedPcrMetrics.jar VALIDATION_STRINGENCY= SILENT AMPLICON_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed INPUT= \u002Fdev\u002Fstdin OUTPUT= output_file.txt REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n4. command type= :begin\n\tcommand input : ['input_file_type *coordinate*', 'input_directory aligner_output']\n\t1. command name= CollectInsertSizeMetrics,command line= java -Xmx1g -jar  \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectInsertSizeMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true I= file_based_input O= output_file.txt H= output_file.pdf TMP_DIR= \u002Fscratch\u002F$USER HISTOGRAM_WIDTH= 500\n5. command type= :begin\n\tcommand input : ['input_file_type *coordinate*', 'input_directory aligner_output']\n\t1. command name= MarkDuplicates,command line= java -Xmx1g -jar  \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FMarkDuplicates.jar VALIDATION_STRINGENCY= SILENT TMP_DIR= \u002Fscratch\u002F$USER MAX_RECORDS_IN_RAM= 1000000 MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP= 2000 ASSUME_SORTED= true I= file_based_input O= output_file.bam METRICS_FILE= output_file.txt\n6. command type= :begin\n\tcommand input : ['input_file_type *coordinate*', 'input_directory aligner_output']\n\t1. command name= cufflinks,command line= \u002Fpackages\u002Fcufflinks\u002F2.1.1\u002Fcufflinks  file_based_input -o output_directory -p 12 -G \u002Faccessory_files\u002Fhuman-ucsc-refGene.gtf\n-------------------------------------------------------------------------------------------------------------\n\n******************* YAP CHECK SUMMARY *******************\n* --Syntax check          : Passed                      *\n* --Compatibility check   : Passed                      *\n* --File paths check      : Passed With Warnings        *\n*********************************************************\n* YAP Configuration overall check status: Passed With Warnings\u003C\u002Fspan\u003E\n\n--------------------------------------- YAP Check Error\u002FWarning Info ---------------------------------------\n-------------------------------------------------------------------------------------------------------------\n--YAP Configuration File paths check status: Passed With Warnings\nWarning: At Line: 21 in file: bowtie_1.0.0_configuration.cfg. Files were found using basename in \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Findexes\u002Fbowtie\u002Fhg19. Please make sure that command: bowtie can work with basenames.\n\n\n-------------------------------------------------------------------------------------------------------------\n-------------------------- YAP configurations check end for Workflow = yap2.3_test --------------------------\n-------------------- PROVENANCE --------------------\n\nPREPROCESS:\n\n\t\u002Fpackages\u002FFastQScreen\u002Fv0.4.1\u002Ffastq_screen --subset 500000 --paired --outdir \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpreprocess_output --conf fastq_screen_v0.4.1.conf --aligner bowtie \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq\n\n\t\u002Fpackages\u002Ffastqc\u002F0.10.1\u002Ffastqc --outdir \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpreprocess_output --extract --threads 12 \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq\n\nALIGNMENT :\n\n\tINPUT: \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq and \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq chunk number= 0\n\n\t\u002Fpackages\u002Fbowtie\u002F1.0.0\u002Fbowtie  \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Findexes\u002Fbowtie\u002Fhg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000003008281_0.797636572311_pipe_0_0_1  -2  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000003008281_0.797636572311_pipe_0_0_2   \u003E\u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000000.sam\n\n\tsamtools view -bhS - samtools sort -on -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000000_queryname | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000000_queryname.sam\n\n\tsamtools view -bhS - samtools sort -o -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000000_coordinate | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000000_coordinate.sam\n\n\tINPUT: \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq and \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq chunk number= 1\n\n\t\u002Fpackages\u002Fbowtie\u002F1.0.0\u002Fbowtie  \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Findexes\u002Fbowtie\u002Fhg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000013008281_0.797636572311_pipe_1_0_1  -2  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000013008281_0.797636572311_pipe_1_0_2   \u003E\u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000001.sam\n\n\tsamtools view -bhS - samtools sort -on -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000001_queryname | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000001_queryname.sam\n\n\tsamtools view -bhS - samtools sort -o -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000001_coordinate | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000001_coordinate.sam\n\n\tINPUT: \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq and \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq chunk number= 2\n\n\t\u002Fpackages\u002Fbowtie\u002F1.0.0\u002Fbowtie  \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Findexes\u002Fbowtie\u002Fhg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000023008281_0.797636572311_pipe_2_0_1  -2  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000023008281_0.797636572311_pipe_2_0_2   \u003E\u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000002.sam\n\n\tsamtools view -bhS - samtools sort -on -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000002_queryname | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000002_queryname.sam\n\n\tsamtools view -bhS - samtools sort -o -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000002_coordinate | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000002_coordinate.sam\n\n\tINPUT: \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq and \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq chunk number= 3\n\n\t\u002Fpackages\u002Fbowtie\u002F1.0.0\u002Fbowtie  \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Findexes\u002Fbowtie\u002Fhg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000033008281_0.797636572311_pipe_3_0_1  -2  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000033008281_0.797636572311_pipe_3_0_2   \u003E\u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000003.sam\n\n\tsamtools view -bhS - samtools sort -on -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000003_queryname | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000003_queryname.sam\n\n\tsamtools view -bhS - samtools sort -o -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000003_coordinate | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000003_coordinate.sam\n\n\tINPUT: \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq and \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq chunk number= 4\n\n\t\u002Fpackages\u002Fbowtie\u002F1.0.0\u002Fbowtie  \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Findexes\u002Fbowtie\u002Fhg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000043008281_0.797636572311_pipe_4_0_1  -2  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000043008281_0.797636572311_pipe_4_0_2   \u003E\u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000004.sam\n\n\tsamtools view -bhS - samtools sort -on -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000004_queryname | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000004_queryname.sam\n\n\tsamtools view -bhS - samtools sort -o -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000004_coordinate | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000004_coordinate.sam\n\n\tINPUT: \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_1.fq and \u002Fdb\u002Fyap\u002Fbenchmark\u002Frobin_50_samples\u002FRN0000108D_2.fq chunk number= 5\n\n\t\u002Fpackages\u002Fbowtie\u002F1.0.0\u002Fbowtie  \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Findexes\u002Fbowtie\u002Fhg19 -q -v 2 -k 10 -m 10 --best -S -p 8 -1  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000053008281_0.797636572311_pipe_5_0_1  -2  \u002Fscratch\u002Fkulkatr1\u002F\u002Fkulkatr1\u002Fyap_temp\u002Faligner_RN0000108D_1_0000053008281_0.797636572311_pipe_5_0_2   \u003E\u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000005.sam\n\n\tsamtools view -bhS - samtools sort -on -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000005_queryname | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000005_queryname.sam\n\n\tsamtools view -bhS - samtools sort -o -m 100000000 - \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000005_coordinate | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000005_coordinate.sam\n\nMERGE ALIGNMENT OUTPUT :\n\n\tsamtools merge -n  -  \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000002_queryname.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000003_queryname.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000005_queryname.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000000_queryname.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000001_queryname.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000004_queryname.bam  | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_queryname.sam\n\n\tsamtools merge  -  \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000005_coordinate.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000001_coordinate.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000000_coordinate.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000002_coordinate.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000003_coordinate.bam \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002Faligner_RN0000108D_1_000004_coordinate.bam  | samtools view -h - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam\n\nPOSTPROCESS :\n\n\tINPUT: \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_queryname.sam\n\n\t\u002Fpackages\u002Fpython\u002F2.6.5_gnu\u002Fbin\u002Fhtseq-count -s no -q  \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_queryname.sam  \u002Faccessory_files\u002Fhuman-ucsc-refGene.gtf  \u003E\u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_htseq-count.out\n\n\tINPUT: \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam\n\n\tyap_exon_count -f 1.0 -exon_coordinates_file \u002Faccessory_files\u002Fhuman-ucsc-final_exon_coord.bed -exon_CoordToNumber_file \u002Faccessory_files\u002Fhuman-ucsc-final_exon_coord_number.bed -i - -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_yap_exon_count\n\n\tjava -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectAlignmentSummaryMetrics.jar VALIDATION_STRINGENCY= SILENT I= \u002Fdev\u002Fstdin O= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectAlignmentSummaryMetrics.txt IS_BISULFITE_SEQUENCED= true ASSUME_SORTED= True REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa QUIET= True\n\n\tjava -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FQualityScoreDistribution.jar VALIDATION_STRINGENCY= SILENT I= \u002Fdev\u002Fstdin O= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_QualityScoreDistribution.txt ASSUME_SORTED= true REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa CHART= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_QualityScoreDistribution.pdf ALIGNED_READS_ONLY= true\n\n\tjava -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FMeanQualityByCycle.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa I= \u002Fdev\u002Fstdin O= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_MeanQualityByCycle.txt CHART= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_MeanQualityByCycle.pdf ALIGNED_READS_ONLY= true\n\n\tjava -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectGcBiasMetrics.jar VALIDATION_STRINGENCY= SILENT I= \u002Fdev\u002Fstdin O= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectGcBiasMetrics.txt SUMMARY_OUTPUT= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectGcBiasMetrics_summary.txt CHART= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectGcBiasMetrics.pdf ASSUME_SORTED= true REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n\n\tjava -Xmx1g -jar  \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectRnaSeqMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true REF_FLAT= \u002Fdb\u002Fyap\u002Fucsc\u002Fmay_02_2013\u002Fgtf\u002Fhg19\u002Fhuman_refflat_for_picard.gff RIBOSOMAL_INTERVALS= \u002Fdb\u002Fyap\u002Fucsc\u002Fmay_02_2013\u002Fgtf\u002Fhg19\u002FHomo_sapiens_assembly19.rRNA.interval_list STRAND_SPECIFICITY= NONE I= \u002Fdev\u002Fstdin O= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectRnaSeqMetrics.txt CHART_OUTPUT= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectRnaSeqMetrics.pdf REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n\n\t\u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCalculateHsMetrics.jar VALIDATION_STRINGENCY= SILENT BAIT_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed INPUT= \u002Fdev\u002Fstdin OUTPUT= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CalculateHsMetrics.txt REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n\n\tjava -Xmx1g -jar \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectTargetedPcrMetrics.jar VALIDATION_STRINGENCY= SILENT AMPLICON_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed TARGET_INTERVALS= \u002Faccessory_files\u002FTruSeq_exome_targeted_regions_for_picard.bed INPUT= \u002Fdev\u002Fstdin OUTPUT= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectTargetedPcrMetrics.txt REFERENCE_SEQUENCE= \u002Fdb\u002Fnibrgenome\u002FNG00006.0\u002Ffasta\u002Fhg19.fa\n\n\tINPUT: \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam\n\n\tjava -Xmx1g -jar  \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FCollectInsertSizeMetrics.jar VALIDATION_STRINGENCY= SILENT ASSUME_SORTED= true I= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam O= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectInsertSizeMetrics.txt H= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_CollectInsertSizeMetrics.pdf TMP_DIR= \u002Fscratch\u002F$USER HISTOGRAM_WIDTH= 500\n\n\tINPUT: \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam\n\n\tjava -Xmx1g -jar  \u002Fpackages\u002Fpicard-tools\u002F1.89\u002FMarkDuplicates.jar VALIDATION_STRINGENCY= SILENT TMP_DIR= \u002Fscratch\u002F$USER MAX_RECORDS_IN_RAM= 1000000 MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP= 2000 ASSUME_SORTED= true I= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam O= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_MarkDuplicates.bam METRICS_FILE= \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output\u002FRN0000108D_1_MarkDuplicates.txt\n\n\tINPUT: \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam\n\n\t\u002Fpackages\u002Fcufflinks\u002F2.1.1\u002Fcufflinks  \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Faligner_output\u002FRN0000108D_1_coordinate.sam -o \u002Ftest_output\u002Fyap2.3_test\u002FRN0000108D_1\u002Fno_barcode_specified\u002Fpostprocess_output -p 12 -G \u002Faccessory_files\u002Fhuman-ucsc-refGene.gtf\n\n--------------------Analysis End Time For Workflow : yap2.3_test 2014\u002F09\u002F04 19:43:14--------------------\u003C\u002Fpre\u003E\n      \u003C\u002Fcode\u003E\n    \u003C\u002Fdiv\u003E\n  \u003C\u002Fdiv\u003E\n\u003C\u002Fdiv\u003E\n\n\u003Ch2\u003E\n\u003Ca name=\"user-content-requirements\" class=\"anchor\" href=\"#requirements\" aria-hidden=\"true\"\u003E\u003Cspan class=\"octicon octicon-link\"\u003E\u003C\u002Fspan\u003E\u003C\u002Fa\u003ERequirements\u003C\u002Fh2\u003E\n\n\u003Cp\u003E\u003Cstrong\u003EYAP only runs on Linux systems!\u003C\u002Fstrong\u003E\u003C\u002Fp\u003E\n\n\u003Ch3\u003EDependencies: \u003C\u002Fh3\u003E\n\n\u003Cp\u003EThe following dependencies have to be first installed in your environment. Once installed, make sure these dependencies are added to your path.\u003C\u002Fp\u003E\n\n\u003Cul class=\"task-list\"\u003E\n\u003Cli\u003ERecent versions of gcc (gcc 4.8.x is well tested)\u003C\u002Fli\u003E\n\u003Cli\u003EPython 2.7.7\u003C\u002Fli\u003E\n\u003Cli\u003EOpenmpi 1.6.5\u003C\u002Fli\u003E\n\u003Cli\u003EPython modules:\n\n\u003Cul class=\"task-list\"\u003E\n\u003Cli\u003EMPI4py - 1.3\u003C\u002Fli\u003E\n\u003Cli\u003EPyPdf - 1.13\u003C\u002Fli\u003E\n\u003Cli\u003ENumpy - 1.7.1\u003C\u002Fli\u003E\n\u003Cli\u003Enetsa-utils - 1.4.3\u003C\u002Fli\u003E\n\u003C\u002Ful\u003E\n\u003C\u002Fli\u003E\n\u003Cli\u003Ebedtools - 2.15.0\u003C\u002Fli\u003E\n\u003Cli\u003Esamtools - 0.1.18\u003C\u002Fli\u003E\n\u003C\u002Ful\u003E\n\n\u003Ch3\u003ESystem Configuration:\u003C\u002Fh3\u003E\n\n\u003Cp\u003E   YAP provides a framework to run external tools and data, so the tools used\nin the workflows drive the system requirements. It can be installed on multicore\nlinux workstation with a decent amount of memory for small data, or on large\ncluster systems to scale optimally for large data processing. The framework has\nbeen tested extensively for \u003Cabbr title=\"Next-Generation Sequencing\"\u003ENGS\u003C\u002Fabbr\u003E\ndata on clusters with minimum system configuration of 8-12 cores and 24-48 GB\nmemory.   \u003C\u002Fp\u003E\n\n\u003Ch2\u003EYAP Setup\u003C\u002Fh2\u003E\n\n\u003Cul class=\"task-list\"\u003E\n\u003Cli\u003EDownload the yap source from \u003Ca href=\"https:\u002F\u002Fgithub.com\u002FNovartis\u002Fyap\u002Farchive\u002Fmaster.zip\"\u003Ehere\u003C\u002Fa\u003E\n\u003C\u002Fli\u003E\n\u003Cli\u003E\n\u003Cp\u003EUncompress the source directory \u003C\u002Fp\u003E\n\n\u003Cp\u003Efor example: uncompress the directory as \u003Ccode\u003E\u002Fhome\u002Fpackages\u002FYAP\u003C\u002Fcode\u003E\u003C\u002Fp\u003E\n\u003C\u002Fli\u003E\n\u003Cli\u003E\n\u003Cp\u003ESet \u003Ccode\u003EYAP_HOME\u003C\u002Fcode\u003E environment variable to the source directory.\u003C\u002Fp\u003E\n\n\u003Cp\u003E\u003Cpre\u003E\u003Ccode\u003E$ export YAP_HOME=\u002Fhome\u002Fpackages\u002FYAP\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\u003C\u002Fp\u003E\n\u003C\u002Fli\u003E\n\u003Cli\u003E\n\u003Cp\u003EAdd bin directory to path\u003C\u002Fp\u003E\n\n\u003Cpre\u003E\u003Ccode\u003E$ export PATH=$PATH:$YAP_HOME\u002Fbin\u002F\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003C\u002Fli\u003E\n\u003Cli\u003E\n\u003Cp\u003ESet \u003Ccode\u003EYAP_LOCAL_TEMPDIR\u003C\u002Fcode\u003E\nenvironment variable for temporary computation. For optimum performance point\nthis directory to a location which is  local to the machine. \u003C\u002Fp\u003E\n\n\u003Cpre\u003E\u003Ccode\u003E$ export YAP_LOCAL_TEMPDIR=\u002Fscratch\u002Fusername\u002Fyap_temp\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003C\u002Fli\u003E\n\u003C\u002Ful\u003E\u003Cp\u003E\u003Cstrong\u003EVerification\u003C\u002Fstrong\u003E\u003C\u002Fp\u003E\n\n\u003Cpre\u003E\u003Ccode\u003E$ echo $YAP_HOME\n    \u002Fhome\u002Fpackages\u002FYAP\n\n$ echo $YAP_LOCAL_TEMPDIR\n    \u002Fscratch\u002Fusername\u002Fyap_temp\n\n$ which yap\n    \u002Fhome\u002Fpackages\u002FYAP\u002Fbin\u002Fyap\n\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\n\u003Ch1\u003E \u003Ca name=\"user-content-running-a-yap-job\" class=\"anchor\"\nhref=\"#running-a-yap-job\" aria-hidden=\"true\"\u003E\u003Cspan class=\"octicon\nocticon-link\"\u003E\u003C\u002Fspan\u003E\u003C\u002Fa\u003ERunning a YAP job\u003C\u002Fh1\u003E\n\n\u003Cp\u003EOnce you've set your environment, it is best to run a quick demo job to get\nthe feel of running YAP. The following section is meant to be interactive and\nhence you would need Linux account access and access to the cluster.\u003C\u002Fp\u003E\n\n\u003Cp\u003EAfter downloading the project, please see the demo configuration files in\n\u003Ccode\u003Eyap\u002Fcfg\u003C\u002Fcode\u003E.\u003C\u002Fp\u003E\n\n\u003Cp\u003EThere are 3 stages in YAP - Preprocess, Alignment and Postprocess. You can\nhave command level control of these three stages in the namesake configuration\nfiles and a workflow level control in the\n\u003Ccode\u003Eworkflow_configuration.\u003C\u002Fcode\u003E\u003C\u002Fp\u003E\n\u003Ctable class=\"table table-striped\"\u003E\n\u003Cthead\u003E\u003Ctr\u003E\n\u003Cth\u003EConfiguration\u003C\u002Fth\u003E\n\u003Cth\u003EPurpose\u003C\u002Fth\u003E\n\u003C\u002Ftr\u003E\u003C\u002Fthead\u003E\n\u003Ctbody\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Ealigner_configuration\u003C\u002Ftd\u003E\n\u003Ctd\u003Ebwa, bowtie, bowtie2, tophat or insert your own aligner\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Epostprocess_configuration\u003C\u002Ftd\u003E\n\u003Ctd\u003Epostalignment packages, generate counts or metrics\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Epreprocess_configuration\u003C\u002Ftd\u003E\n\u003Ctd\u003Epre-alignment packages to massage your seqdata\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Eworkflow_configuration\u003C\u002Ftd\u003E\n\u003Ctd\u003Emanage metadata, specify input files, paths and output directories\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Eyap_sge\u003C\u002Ftd\u003E\n\u003Ctd\u003Esubmitting your job to the cluster\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Ftbody\u003E\n\u003C\u002Ftable\u003E\u003Cp\u003EThe demo runs a RNASeq QC and counts workflow consisting of:\u003C\u002Fp\u003E\n\n\u003Cul class=\"task-list\"\u003E\n\u003Cli\u003EPreprocess: FastQC, Fastqscreen\u003C\u002Fli\u003E\n\u003Cli\u003EAlignment: Bowtie, both queryname and coordinate sorted\u003C\u002Fli\u003E\n\u003Cli\u003EPostprocess: yap junction and exon counts, Picard tools (PostQC), HTSeq (Raw counts) and Cufflinks (normalized counts)\u003C\u002Fli\u003E\n\u003C\u002Ful\u003E\u003Cp\u003EWe run this workflow on 2 nodes on the UGE cluster.\u003C\u002Fp\u003E\n\n\u003Cp\u003ETo run the yap_demo job, we next need to check to see if our configuration files are correct using the command.\u003C\u002Fp\u003E\n\n\u003Cpre\u003E\u003Ccode\u003Ecd &lt;your_working directory&gt;\nyap --check workflow_configuration.cfg\n\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\n\u003Cp\u003EThe \u003Ccode\u003Eyap --check\u003C\u002Fcode\u003E  command checks to see\u003C\u002Fp\u003E\n\n\u003Cul class=\"task-list\"\u003E\n\u003Cli\u003EIf all paths specified are valid\u003C\u002Fli\u003E\n\u003Cli\u003EIf YAP finds the appropriate input files\u003C\u002Fli\u003E\n\u003Cli\u003EChecks for syntax errors\u003C\u002Fli\u003E\n\u003Cli\u003ELists commands to be executed.\u003C\u002Fli\u003E\n\u003Cli\u003EGives section-wise error\u002Fwarning report.\u003C\u002Fli\u003E\n\u003C\u002Ful\u003E\n\n\u003Ch2\u003ERunning the YAP job\u003C\u002Fh2\u003E\n\n\u003Cpre\u003E\u003Ccode\u003Empirun -n &lt;number_of_cores&gt; yap workflow_configuration.cfg\n\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\n\u003Cp\u003EIf you have a SGE environment, pass the number of slots into the \u003Ccode\u003E$NSLOTS\u003C\u002Fcode\u003E variable.\u003C\u002Fp\u003E\n\u003Cscript\u003E\n$('abbr').tooltip();\n\u003C\u002Fscript\u003E\n"}}}("yap","Yet Another Pipeline"))],session:{basepath:"",forArtifactory:false}};if('serviceWorker' in navigator)navigator.serviceWorker.register('/service-worker.js');</script><script src=/client/1ef9048e76cf74561a9e/main.js></script> 