<!DOCTYPE html> <head> <meta charset=utf-8> <meta content="width=device-width,initial-scale=1" name=viewport> <meta content=#333333 name=theme-color> <base href=/ > <link href=global.css rel=stylesheet> <style>main.svelte-1pnr8ds{position:relative;box-sizing:border-box;min-height:100vh;width:100%;padding-top:3.5rem;padding-bottom:2.75rem}main.homepage.svelte-1pnr8ds{padding-right:2rem;padding-left:2rem}.bg.contentPage.svelte-1pnr8ds{padding-left:1rem;padding-right:1rem}@media screen and (min-width:775px){.bg.contentPage.svelte-1pnr8ds{padding-left:calc(75px + 1rem);padding-right:4rem}}.bg.svelte-1pnr8ds{background-size:auto,0;background-repeat:auto,no-repeat}@media screen and (min-width:775px){.bg.svelte-1pnr8ds{background-position:right,left;background-attachment:fixed,fixed;background-repeat:auto,repeat-y;background-size:cover,120px}}a{font-weight:700}nav.svelte-2cyc0n{z-index:20;position:fixed;top:0;left:0;right:0;background-color:#fff;font-weight:300;padding:0 1em 1px;transition:box-shadow .2s;margin:0;display:flex;flex-flow:row nowrap}nav.elevated.svelte-2cyc0n{box-shadow:0 0 .5rem rgba(0,0,0,.3),0 1px #eee}.brand-logo.svelte-2cyc0n{height:1em;transform:scale(1) translateX(0);transition:transform .6s ease-in-out}ul.svelte-2cyc0n{margin:0;padding:0;flex-shrink:0}ul.svelte-2cyc0n:first-child{flex-grow:1}ul.svelte-2cyc0n::after{content:'';display:block;clear:both}li.svelte-2cyc0n{display:block;float:left}.selected.svelte-2cyc0n{position:relative;display:inline-block;font-weight:700}.selected.svelte-2cyc0n::after{position:absolute;content:'';width:calc(100% - 1em);height:2px;background-color:#0460a9;display:block;bottom:-1px}a.svelte-2cyc0n{text-decoration:none;padding:1em .5em;display:block}.summary.svelte-dzuyvs{margin:0 0 1em;padding:1em;border-top:3px solid #000;transition:transform .2s;background-color:#fff;box-shadow:-1em 0 #fff}@media screen and (min-width:775px){.summary.svelte-dzuyvs{float:right;width:20rem;transform:translateX(1rem)}}@media screen and (min-width:1200px){.summary.svelte-dzuyvs{float:right;width:20rem;transform:translateX(3rem)}}.content.svelte-dzuyvs h1,h2,h3,h4,h5,h6{margin:0 0 .5em 0;line-height:1.2}.content.svelte-dzuyvs h1,h1.svelte-dzuyvs{font-weight:700;font-size:3rem;margin-bottom:3rem}.content.svelte-dzuyvs h2{font-size:2rem;font-weight:400}.content.svelte-dzuyvs :not(h1):not(hr)+h2{border-top:1px solid #000;padding-top:1em;margin-top:1em}.content.svelte-dzuyvs h3{color:#0460a9}.content.svelte-dzuyvs pre{background-color:#f9f9f9;box-shadow:inset 1px 1px 5px rgba(0,0,0,.05);padding:.5em;border-radius:2px;overflow-x:auto}.content.svelte-dzuyvs pre code{background-color:transparent;padding:0}.content.svelte-dzuyvs ul{line-height:1.5}.content.svelte-dzuyvs li{margin:0 0 .5em 0}.content.svelte-dzuyvs img{max-width:100%}.content.svelte-dzuyvs table{margin:1em 2em;border-collapse:collapse}.content.svelte-dzuyvs th{text-align:left;padding:0 .5em}.content.svelte-dzuyvs thead th{color:#0460a9;font-weight:700;text-transform:uppercase;border-bottom:1px solid #0460a9}.content.svelte-dzuyvs td{text-align:left;padding:.25em .5em}.callout-wrapper.svelte-dzuyvs>div{margin-top:4rem;margin-left:-1rem;margin-right:-1rem}@media screen and (min-width:775px){.callout-wrapper.svelte-dzuyvs>div{margin-left:calc(-75px - 3rem);margin-right:-4rem}}.btn.svelte-zemmdc,a[role=button]{border:1px solid #000;background-color:#fff;color:#000;transition:color .2s,background-color .2s;font-weight:700;display:inline-flex;align-items:center;justify-content:center;text-decoration:none;padding:.25em .75em;white-space:nowrap;cursor:pointer}.btn.svelte-zemmdc:hover,a[role=button]:hover{background-color:#000;color:#fff}.btn.primary.svelte-zemmdc{border-color:#fff;background-color:#0460a9;color:#fff}.btn.primary.svelte-zemmdc:hover{background-color:#fff;color:#0460a9}.callout.svelte-bf2rc0{background-color:#0460a9;color:#fff;padding:2rem 1rem}@media screen and (min-width:775px){.callout.svelte-bf2rc0{padding-left:calc(75px + 3rem)}}.callout-content.svelte-bf2rc0{max-width:50rem}footer.svelte-1o8yfks{display:flex;flex-flow:row nowrap;font-size:.75rem;padding:.5rem .75rem;align-items:center;border-top:1px solid #eee;position:fixed;bottom:0;left:0;right:0;background-color:#fff}@media screen and (max-width:499px){footer.svelte-1o8yfks{text-align:center;flex-flow:column nowrap}}.footer-left.svelte-1o8yfks{flex-grow:1}.footer-right.svelte-1o8yfks{flex-grow:0}a.svelte-1o8yfks{color:rgba(0,0,0,.5);text-decoration:none;font-weight:400}a.svelte-1o8yfks:hover{text-decoration:underline}.social.svelte-1o8yfks{font-size:.825rem}.social.svelte-1o8yfks a.svelte-1o8yfks{border:1px solid #000;background-color:#fff;color:#000;transition:color .2s,background-color .2s;width:1.5rem;height:1.5rem;display:inline-flex;align-items:center;justify-content:center;text-decoration:none;margin:0 .125rem}.social.svelte-1o8yfks a.svelte-1o8yfks:hover{background-color:#000;color:#fff}.sep.svelte-1o8yfks::before{content:'|'}</style> <noscript id=sapper-head-start></noscript><link href=/favicon.ico rel=icon type=image/ico><link href=/manifest.json rel=manifest><title>Peax</title><noscript id=sapper-head-end></noscript> <link href=/client/1ef9048e76cf74561a9e/main.js rel=preload as=script><link href=/client/1ef9048e76cf74561a9e/projects_$slug.4.js rel=preload as=script></head> <body> <div id=sapper> <nav class=svelte-2cyc0n> <ul class=svelte-2cyc0n> <li class=svelte-2cyc0n> <a href="" class=svelte-2cyc0n>NIBR Open Source</a> </li> </ul> <ul class=svelte-2cyc0n> <li class=svelte-2cyc0n> <a href=https://www.novartis.com class=svelte-2cyc0n rel="nofollow noreferrer" target=_blank> <img alt="Novartis logo" src=/novartis-logo.png class="svelte-2cyc0n brand-logo"> </a> </li> </ul> </nav> <div class="contentPage svelte-1pnr8ds bg" style="background-image:linear-gradient(to right,rgba(255,255,255,0),rgba(255,255,255,0) 75px,rgba(255,255,255,255) 75px,rgba(255,255,255,255)),url(/blue-carbon.png)"> <main class="contentPage svelte-1pnr8ds"> <h1 class=svelte-dzuyvs>Peax</h1> <div class="svelte-dzuyvs summary"> <h2>Project details</h2> <p>Released 9/15/2018</p> <p><p>Peax is a tool for interactive concept learning and exploration of epigenomic patterns based on unsupervised machine learning with autoencoders.</p> <p></p> <a href=https://github.com/novartis/peax class=svelte-zemmdc class=btn rel="nofollow noreferrer" target=_blank component=a> GitHub repo </a> </div> <div class="svelte-dzuyvs content"> <blockquote> <p>A pattern explorer for epigenomic data.</p> </blockquote> <p><img alt="Peax's UI" src=/projects/peax/teaser.png></p> <blockquote> <p>Epigenomic data expresses a rich body of diverse patterns that help to identify regulatory elements like promoter, enhancers, etc. But finding these patterns reliably genome wide is challenging. Peax is a tool for interactive visual pattern search and exploration of epigenomic patterns based on unsupervised representation learning with convolutional autoencoders. The visual search is driven by manually labeled genomic regions for actively learning a classifier to reflect your notion of interestingness. More at <a href=http://peax.lekschas.de>peax.lekschas.de</a>.</p> </blockquote> <h2 id=installation>Installation</h2> <pre><code class=language-bash>git clone https://github.com/Novartis/peax peax && cd peax
make install</code></pre> <p><em>Do not fear, <code>make install</code> is just a convenience function for setting up conda and installing npm packages.</em></p> <p><strong>Notes:</strong></p> <ul> <li>If you're a macOS user you might need to <a href=https://brew.sh>brew</a> install <code>libpng</code> and <code>openssl</code> for the <a href=https://github.com/nvictus/pybbi>pybbi</a> package (see <a href=https://github.com/nvictus/pybbi/issues/2>here</a>) and <code>xz</code> for pysam (if you see an error related to <code>lzma.h</code>).</li> </ul> <h2 id=overview>Overview</h2> <p>Peax consists of three main parts:</p> <ol> <li>A server application for serving genomic and autoencoded data on the web. [<a href=https://github.com/Novartis/peax/tree/develop/server>/server</a>].</li> <li>A user interface for exploring, visualizing, and interactively labeling genomic regions. [<a href=https://github.com/Novartis/peax/tree/develop/ui>/ui</a>].</li> <li>A set of examples showing how to configure Peax and build your own. [<a href=https://github.com/Novartis/peax/tree/develop/examples>/examples</a>]</li> </ol> <h2 id=data>Data</h2> <p>We provide 6 autoencoders trained on 3 kb, 12 kb, and 120 kb window sizes (with 25, 100, and 1000 bp binning) on DNase-seq and histone mark ChIP-seq data.</p> <p>You can find the autoencoder at <a href=https://zenodo.org/record/2609763>zenodo.org/record/2609763</a>.</p> <h2 id=preprint>Preprint</h2> <p>Lekschas et al., <a href=https://www.biorxiv.org/content/10.1101/597518v1>Peax: Interactive Visual Pattern Search in Sequential Data Using Unsupervised Deep Representation Learning</a>, <em>bioRxiv</em>, 2019, doi: <a href=10.1101/597518>10.1101/597518</a>.</p> <h2 id=quick-start>Quick start</h2> <p>Peax comes with <a href=#autoencoders>6 autoencoders</a> for DNase-seq and histone mark ChIP-seq data and several example configurations for which we provide convenience scripts to get you started as quickly as possible.</p> <p>For instance, run one of the following commands to start Peax with a DNase-seq track for 3 kb, 12 kb, and 120 kb genomic windows.</p> <table> <thead> <tr> <th>Command</th> <th>Window Size</th> <th>Step Freq.</th> <th>Chromosomes</th> </tr> </thead> <tr> <td><code>make example-3kb</code></td> <td>3 kb</td> <td>2</td> <td>21</td> </tr> <tr> <td><code>make example-12kb</code></td> <td>12 kb</td> <td>3</td> <td>20-21</td> </tr> <tr> <td><code>make example-120kb</code></td> <td>120 kb</td> <td>6</td> <td>17-21</td> </tr> </table> <p><strong>Note:</strong> The first time Peax is started it will precompute the datasets for exploration. This can take a few minutes depending on your hardware. Also, these demos will only prepare the above mentioned chromosomes, so don't try to search for patterns on another chromosome. It won't work! For your own data you can freely configure this of course.</p> <p>The scripts will download test ENCODE tracks and use the matching configuration to start the server. More examples are described in <a href=https://github.com/Novartis/peax/tree/develop/examples><code>/examples</code></a>.</p> <h2 id=get-started>Get Started</h2> <p>In the following we describe how you can configure Peax for your own data.</p> <h4 id=configure-peax-with-your-data>Configure Peax with your data</h4> <p>Next, you need to configure Peax with your data to tell it which tracks you want to visualize in HiGlass and which of those tracks are encodable using an (auto)encoder.</p> <p>The fastest way to get started is to copy the example config:</p> <pre><code>cp config.json.sample config.json</code></pre><p>The config file has 6 top level properties:</p> <table> <thead> <tr> <th>Field</th> <th>Description</th> <th>Dtype</th> </tr> </thead> <tr> <td>encoders</td> <td>List of encoders.</td> <td>list</td> </tr> <tr> <td>datasets</td> <td>List of tracks.</td> <td>list</td> </tr> <tr> <td>coords</td> <td>Genome coordinates. Peax currently supports hg19, hg28, mm9, and mm10</td> <td>str</td> </tr> <tr> <td>chroms</td> <td>Chromosomes to to be searched. If omitted all chromosomes will be prepared for searching.</td> <td>list</td> </tr> <tr> <td>step_freq</td> <td>Step frequency of the sliding window approach. E.g., given an encoder with window size 12 kb, a step frequency of 6 means that every 2 kb a 12 kb window will be extracted from the bigWig.</td> <td>int</td> </tr> <tr> <td>db_path</td> <td>Relative path to the sqlite db for storing searches.</td> <td>str</td> </tr> </table> <p>The main parts to adjust are <code>encoders</code> and <code>datasets</code>. <code>encoders</code> is a list of (auto)encoder definitions for different datatypes.T here are two ways to configure an (auto)encoder: (a) point to a pre-defined autoencoder or (b) configure from scratch.</p> <p>Assuming you want to use a predefined autoencoder all you have to do is</p> <table> <thead> <tr> <th>Field</th> <th>Description</th> <th>Dtype</th> </tr> </thead> <tr> <td>content_type</td> <td>Unique string identifying the autoencoder in the configuration file</td> <td>str</td> </tr> <tr> <td>from_file</td> <td>Relative path to the encoder configuration file.</td> <td>str</td> </tr> </table> <p><strong>Example:</strong></p> <pre><code class=language-json>{
  "content_type": "histone-mark-chip-seq-3kb",
  "from_file": "examples/encoders.json"
}</code></pre> <p>The encoder configuration file is a dictionary with the top level keys acting as the identifier and need to match <code>content_type</code> above. Given the example from above the file could look like this:</p> <pre><code class=language-json>{
  "histone-mark-chip-seq-3kb": {},
  "dnase-seq-3kb": {}
}</code></pre> <p>See <code>[encoders.json](encoders.json)</code> for an example. The specific definition if an autoencoder is the same as described in the following.</p> <p>To configure an autoencoder from scratch you need to provide a dictionary with the following required format:</p> <table> <thead> <tr> <th>Field</th> <th>Description</th> <th>Defaults</th> <th>Dtype</th> </tr> </thead> <tr> <td>autoencoder</td> <td>Relative path to your pickled autoencoder model. (hdf5 file)</td> <td> </td> <td>str</td> </tr> <tr> <td>encoder</td> <td>Relative path to your pickled encoder model. (hdf5 file)</td> <td> </td> <td>str</td> </tr> <tr> <td>decoder</td> <td>Relative path to your pickled decoder model. (hdf5 file)</td> <td> </td> <td>str</td> </tr> <tr> <td>content_type</td> <td>Unique string describing the content this autoencoder can handle. Data tracks with the same content type will be encoded by this autoencoder.</td> <td> </td> <td>str</td> </tr> <tr> <td>window_size</td> <td>Window size in base pairs used for training the autoencoder.</td> <td> </td> <td>int</td> </tr> <tr> <td>resolution</td> <td>Resolution or bin size of the window in base pairs.</td> <td> </td> <td>int</td> </tr> <tr> <td>latent_dim</td> <td>Number of latent dimensions of the encoded windows.</td> <td> </td> <td>int</td> </tr> <tr> <td>input_dim</td> <td>Number of input dimensions for Keras. For 1D data these are 3: samples, data length (which is <code>window_size</code> / <code>resolution</code>), channels.</td> <td>3</td> <td>int</td> </tr> <tr> <td>channels</td> <td>Number of channels of the input data. This is normally 1.</td> <td>1</td> <td>int</td> </tr> </table> <p><em>Note that if you have specified an <code>autoencoder</code> you do not need to provide separate <code>encoder</code> and <code>decoder</code> models.</em></p> <p><strong>Example:</strong></p> <pre><code class=language-json>{
  "encoder": "path/to/my-12kb-chip-seq-encoder.h5",
  "decoder": "path/to/my-12kb-chip-seq-decoder.h5",
  "content_type": "histone-mark-chip-seq",
  "window_size": 12000,
  "resolution": 100,
  "channels": 1,
  "input_dim": 3,
  "latent_dim": 12
}</code></pre> <p>Datasets require the following format:</p> <table> <thead> <tr> <th>Field</th> <th>Description</th> <th>Dtype</th> </tr> </thead> <tr> <td>filepath</td> <td>Relative path to the data file (bigWig or bigBed).</td> <td>str</td> </tr> <tr> <td>content_type</td> <td>Unique string describing the content this dataset. If you want to search for patterns in this track you need to have an autoencoder with a matching content type.</td> <td>str</td> </tr> <tr> <td>id</td> <td>A unique string identifying your track. (Optional)</td> <td>str</td> </tr> <tr> <td>name</td> <td>A human readable name to be shown in HiGlass. (Optional)</td> <td>str</td> </tr> </table> <p><strong>Example:</strong></p> <pre><code class=language-json>{
  "filepath": "data/chip-seq/my-fancy-gm12878-chip-seq-h3k27ac-fc-signal.bigWig",
  "content_type": "histone-mark-chip-seq",
  "uuid": "my-fancy-gm12878-chip-seq-h3k27c-track",
  "name": "My Fancy GM12878 ChIP-Seq H3k27c Track"
}</code></pre> <h4 id=start-peax>Start Peax</h4> <p>First, start the Peax server to serve your data.</p> <p><strong>Note:</strong> The first time you run Peax on a new dataset all the data will be prepared! Depending on your machine this can take some time. If you want to track the progress activate the debugging mode using <code>-d</code>.</p> <pre><code class=language-bash>python start.py</code></pre> <p>Now go to <a href=http://localhost:5000>http://localhost:5000</a>.</p> <p>To <code>start.py</code> script supports the following options:</p> <pre><code class=language-bash>usage: start.py [-h] [-c CONFIG] [--clear] [--clear-cache]
                [--clear-cache-at-exit] [--clear-db] [-d] [--host HOST]
                [--port PORT] [-v]

Peak Explorer CLI

optional arguments:
  -h, --help            show this help message and exit
  -c CONFIG, --config CONFIG
                        path to your JSON config file
  --clear               clears the cache and database on startup
  --clear-cache         clears the cache on startup
  --clear-cache-at-exit
                        clear the cache on shutdown
  --clear-db            clears the database on startup
  -d, --debug           turn on debug mode
  --host HOST           customize the hostname
  --port PORT           customize the port
  -v, --verbose         turn verbose logging on</code></pre> <p>The <code>hostname</code> defaults to <code>localhost</code> and the <code>port</code> of the backend server defaults to <code>5000</code>.</p> <p>In order to speed up subsequend user interaction, Peax initially prepapres all the data and caches that data under <code>/cache</code>. You can always remove this directory manually or clear the cache on startup or at exist using the <code>--clear</code> as specified above.</p> <hr> <h2 id=development>Development</h2> <p>Handy commands to keep in mind:</p> <ul> <li><code>make install</code> installs the conda environment and npm packages and builds the UI</li> <li><code>make update</code> updates the conda environment and npm packages and rebuilds the UI</li> <li><code>make build</code> builds the UI</li> <li><code>./start.py</code> starts the Flask server application for serving data</li> <li>[/ui]: <code>npm install</code> installs and updates all the needed packages for the frontend</li> <li>[/ui]: <code>npm build</code> creates the production built of the frontend</li> <li>[/ui]: <code>npm start</code> starts a dev server with hot reloading for the frontend</li> </ul> <p>To start developing on the server and the ui in parallel, first start the backend server application using <code>./start.py</code> and then start the frontend server application from <code>./ui</code> using <code>npm start</code>. Both server's watch the source code, so whenever you make changes to the source code the servers will reload.</p> <h3 id=configuration>Configuration</h3> <p>There are 2 types of configuration files. The <a href=#configure-peax-with-your-data>backend server configuration</a> defines the datasets to explore and is described in detail <a href=#configure-peax-with-your-data>above</a>.</p> <p>Additionally, the frontend application can be configured to talk to a different backend server and port if needed. Get started by copying the example configuration:</p> <pre><code class=language-bash>cd ui && cp config.json.sample config.json</code></pre> <p>By default the <code>server</code> is dynamically set to the hostname of the server running the frontend application. I.e., it is assumed that the backend server application is running on the same host as the frontend application. The <code>port</code> of the server defaults to <code>5000</code>.</p> <h3 id=start-the-backend-and-frontend-apps>Start the backend and frontend apps</h3> <p>For development the backend and frontend applications run as seperate server applications.</p> <pre><code class=language-bash># Backend server
./start.py --debug --config path/to/your/config.json

# Frontend server
cd ui && npm start</code></pre> </div> <div class="svelte-dzuyvs callout-wrapper"> <div class="svelte-bf2rc0 callout"> <div class="svelte-bf2rc0 callout-content"> <h2>Interested in NIBR Engineering?</h2> <p>At NIBR, you'll be at the forefront of technology — helping to shape it, develop it, and make it impactful. Partnering with scientists, our engineers create cutting-edge, state-of-the-art solutions that accelerate drug discovery and ultimately improve patients’ lives.</p> <a href=https://www.novartis.com/careers/careers-research class="primary svelte-zemmdc" class="btn primary" rel="nofollow noopener" target=_blank variant=primary> Learn more </a> </div> </div> </div> </main> </div> <footer class=svelte-1o8yfks> <div class="svelte-1o8yfks footer-left"> © 2019 <a href=https://www.novartis.com class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank>Novartis AG</a> <span class="svelte-1o8yfks sep"></span> <a href=https://www.novartis.com/terms-use class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank> Terms of use </a> <span class="svelte-1o8yfks sep"></span> <a href=https://www.novartis.com/privacy-policy class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank> Privacy policy </a> <span class="svelte-1o8yfks sep"></span> <a href=mailto:open.source@novartis.com class=svelte-1o8yfks> open.source@novartis.com </a> </div> <div class="svelte-1o8yfks footer-right social"> <a href=https://www.github.com/Novartis class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=GitHub> <i class="fab fa-github"></i> </a> <a href=https://www.facebook.com/novartis class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=Facebook> <i class="fab fa-facebook-f"></i> </a> <a href=https://twitter.com/NovartisScience class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=Twitter> <i class="fab fa-twitter"></i> </a> <a href=https://www.linkedin.com/company/novartis-institutes-for-biomedical-research class=svelte-1o8yfks rel="nofollow noreferrer" target=_blank title=LinkedIn> <i class="fab fa-linkedin-in"></i> </a> </div> </footer></div> <script>__SAPPER__={baseUrl:"",preloaded:[void 0,null,(function(a,b){return {project:{slug:b,metadata:{name:a,abbreviation:a,projectName:b,launched:"2018-09-15T00:00:00.000Z",icon:"\u002Fprojects\u002Fpeax\u002Fpeax-icon.png",summary:"Interactive concept learning and visual exploration of epigenomic patterns.",description:"\u003Cp\u003EPeax is a tool for interactive concept learning and exploration of epigenomic patterns based on unsupervised machine learning with autoencoders.\u003C\u002Fp\u003E\n",dateString:"Sat Sep 15 2018",title:a,styles:[],scripts:[]},html:"\u003Cblockquote\u003E\n\u003Cp\u003EA pattern explorer for epigenomic data.\u003C\u002Fp\u003E\n\u003C\u002Fblockquote\u003E\n\u003Cp\u003E\u003Cimg src=\"\u002Fprojects\u002Fpeax\u002Fteaser.png\" alt=\"Peax&#39;s UI\"\u003E\u003C\u002Fp\u003E\n\u003Cblockquote\u003E\n\u003Cp\u003EEpigenomic data expresses a rich body of diverse patterns that help to identify\nregulatory elements like promoter, enhancers, etc. But finding these patterns reliably\ngenome wide is challenging. Peax is a tool for interactive visual pattern search and\nexploration of epigenomic patterns based on unsupervised representation learning with\nconvolutional autoencoders. The visual search is driven by manually labeled genomic\nregions for actively learning a classifier to reflect your notion of interestingness.\nMore at \u003Ca href=\"http:\u002F\u002Fpeax.lekschas.de\"\u003Epeax.lekschas.de\u003C\u002Fa\u003E.\u003C\u002Fp\u003E\n\u003C\u002Fblockquote\u003E\n\u003Ch2 id=\"installation\"\u003EInstallation\u003C\u002Fh2\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-bash\"\u003Egit clone https:\u002F\u002Fgithub.com\u002FNovartis\u002Fpeax peax &amp;&amp; cd peax\nmake install\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003E\u003Cem\u003EDo not fear, \u003Ccode\u003Emake install\u003C\u002Fcode\u003E is just a convenience function for setting up conda and installing npm packages.\u003C\u002Fem\u003E\u003C\u002Fp\u003E\n\u003Cp\u003E\u003Cstrong\u003ENotes:\u003C\u002Fstrong\u003E\u003C\u002Fp\u003E\n\u003Cul\u003E\n\u003Cli\u003EIf you&#39;re a macOS user you might need to \u003Ca href=\"https:\u002F\u002Fbrew.sh\"\u003Ebrew\u003C\u002Fa\u003E install \u003Ccode\u003Elibpng\u003C\u002Fcode\u003E and \u003Ccode\u003Eopenssl\u003C\u002Fcode\u003E for the \u003Ca href=\"https:\u002F\u002Fgithub.com\u002Fnvictus\u002Fpybbi\"\u003Epybbi\u003C\u002Fa\u003E package (see \u003Ca href=\"https:\u002F\u002Fgithub.com\u002Fnvictus\u002Fpybbi\u002Fissues\u002F2\"\u003Ehere\u003C\u002Fa\u003E) and \u003Ccode\u003Exz\u003C\u002Fcode\u003E for pysam (if you see an error related to \u003Ccode\u003Elzma.h\u003C\u002Fcode\u003E).\u003C\u002Fli\u003E\n\u003C\u002Ful\u003E\n\u003Ch2 id=\"overview\"\u003EOverview\u003C\u002Fh2\u003E\n\u003Cp\u003EPeax consists of three main parts:\u003C\u002Fp\u003E\n\u003Col\u003E\n\u003Cli\u003EA server application for serving genomic and autoencoded data on the web. [\u003Ca href=\"https:\u002F\u002Fgithub.com\u002FNovartis\u002Fpeax\u002Ftree\u002Fdevelop\u002Fserver\"\u003E\u002Fserver\u003C\u002Fa\u003E].\u003C\u002Fli\u003E\n\u003Cli\u003EA user interface for exploring, visualizing, and interactively labeling genomic regions. [\u003Ca href=\"https:\u002F\u002Fgithub.com\u002FNovartis\u002Fpeax\u002Ftree\u002Fdevelop\u002Fui\"\u003E\u002Fui\u003C\u002Fa\u003E].\u003C\u002Fli\u003E\n\u003Cli\u003EA set of examples showing how to configure Peax and build your own. [\u003Ca href=\"https:\u002F\u002Fgithub.com\u002FNovartis\u002Fpeax\u002Ftree\u002Fdevelop\u002Fexamples\"\u003E\u002Fexamples\u003C\u002Fa\u003E]\u003C\u002Fli\u003E\n\u003C\u002Fol\u003E\n\u003Ch2 id=\"data\"\u003EData\u003C\u002Fh2\u003E\n\u003Cp\u003EWe provide 6 autoencoders trained on 3 kb, 12 kb, and 120 kb window sizes (with 25,\n100, and 1000 bp binning) on DNase-seq and histone mark ChIP-seq data.\u003C\u002Fp\u003E\n\u003Cp\u003EYou can find the autoencoder at \u003Ca href=\"https:\u002F\u002Fzenodo.org\u002Frecord\u002F2609763\"\u003Ezenodo.org\u002Frecord\u002F2609763\u003C\u002Fa\u003E.\u003C\u002Fp\u003E\n\u003Ch2 id=\"preprint\"\u003EPreprint\u003C\u002Fh2\u003E\n\u003Cp\u003ELekschas et al., \u003Ca href=\"https:\u002F\u002Fwww.biorxiv.org\u002Fcontent\u002F10.1101\u002F597518v1\"\u003EPeax: Interactive Visual Pattern Search in Sequential Data Using Unsupervised Deep Representation Learning\u003C\u002Fa\u003E,\n\u003Cem\u003EbioRxiv\u003C\u002Fem\u003E, 2019, doi: \u003Ca href=\"10.1101\u002F597518\"\u003E10.1101\u002F597518\u003C\u002Fa\u003E.\u003C\u002Fp\u003E\n\u003Ch2 id=\"quick-start\"\u003EQuick start\u003C\u002Fh2\u003E\n\u003Cp\u003EPeax comes with \u003Ca href=\"#autoencoders\"\u003E6 autoencoders\u003C\u002Fa\u003E for DNase-seq and histone mark\nChIP-seq data and several example configurations for which we provide\nconvenience scripts to get you started as quickly as possible.\u003C\u002Fp\u003E\n\u003Cp\u003EFor instance, run one of the following commands to start Peax with a DNase-seq\ntrack for 3 kb, 12 kb, and 120 kb genomic windows.\u003C\u002Fp\u003E\n\u003Ctable\u003E\n\u003Cthead\u003E\n\u003Ctr\u003E\n\u003Cth\u003ECommand\u003C\u002Fth\u003E\n\u003Cth\u003EWindow Size\u003C\u002Fth\u003E\n\u003Cth\u003EStep Freq.\u003C\u002Fth\u003E\n\u003Cth\u003EChromosomes\u003C\u002Fth\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Fthead\u003E\n\u003Ctbody\u003E\u003Ctr\u003E\n\u003Ctd\u003E\u003Ccode\u003Emake example-3kb\u003C\u002Fcode\u003E\u003C\u002Ftd\u003E\n\u003Ctd\u003E3 kb\u003C\u002Ftd\u003E\n\u003Ctd\u003E2\u003C\u002Ftd\u003E\n\u003Ctd\u003E21\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003E\u003Ccode\u003Emake example-12kb\u003C\u002Fcode\u003E\u003C\u002Ftd\u003E\n\u003Ctd\u003E12 kb\u003C\u002Ftd\u003E\n\u003Ctd\u003E3\u003C\u002Ftd\u003E\n\u003Ctd\u003E20-21\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003E\u003Ccode\u003Emake example-120kb\u003C\u002Fcode\u003E\u003C\u002Ftd\u003E\n\u003Ctd\u003E120 kb\u003C\u002Ftd\u003E\n\u003Ctd\u003E6\u003C\u002Ftd\u003E\n\u003Ctd\u003E17-21\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Ftbody\u003E\u003C\u002Ftable\u003E\n\u003Cp\u003E\u003Cstrong\u003ENote:\u003C\u002Fstrong\u003E The first time Peax is started it will precompute the datasets for\nexploration. This can take a few minutes depending on your hardware. Also, these demos\nwill only prepare the above mentioned chromosomes, so don&#39;t try to search for patterns\non another chromosome. It won&#39;t work! For your own data you can freely configure this\nof course.\u003C\u002Fp\u003E\n\u003Cp\u003EThe scripts will download test ENCODE tracks and use the matching\nconfiguration to start the server. More examples are described in \u003Ca href=\"https:\u002F\u002Fgithub.com\u002FNovartis\u002Fpeax\u002Ftree\u002Fdevelop\u002Fexamples\"\u003E\u003Ccode\u003E\u002Fexamples\u003C\u002Fcode\u003E\u003C\u002Fa\u003E.\u003C\u002Fp\u003E\n\u003Ch2 id=\"get-started\"\u003EGet Started\u003C\u002Fh2\u003E\n\u003Cp\u003EIn the following we describe how you can configure Peax for your own data.\u003C\u002Fp\u003E\n\u003Ch4 id=\"configure-peax-with-your-data\"\u003EConfigure Peax with your data\u003C\u002Fh4\u003E\n\u003Cp\u003ENext, you need to configure Peax with your data to tell it which tracks you want to visualize in HiGlass and which of those tracks are encodable using an (auto)encoder.\u003C\u002Fp\u003E\n\u003Cp\u003EThe fastest way to get started is to copy the example config:\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode\u003Ecp config.json.sample config.json\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\u003Cp\u003EThe config file has 6 top level properties:\u003C\u002Fp\u003E\n\u003Ctable\u003E\n\u003Cthead\u003E\n\u003Ctr\u003E\n\u003Cth\u003EField\u003C\u002Fth\u003E\n\u003Cth\u003EDescription\u003C\u002Fth\u003E\n\u003Cth\u003EDtype\u003C\u002Fth\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Fthead\u003E\n\u003Ctbody\u003E\u003Ctr\u003E\n\u003Ctd\u003Eencoders\u003C\u002Ftd\u003E\n\u003Ctd\u003EList of encoders.\u003C\u002Ftd\u003E\n\u003Ctd\u003Elist\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Edatasets\u003C\u002Ftd\u003E\n\u003Ctd\u003EList of tracks.\u003C\u002Ftd\u003E\n\u003Ctd\u003Elist\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Ecoords\u003C\u002Ftd\u003E\n\u003Ctd\u003EGenome coordinates. Peax currently supports hg19, hg28, mm9, and mm10\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Echroms\u003C\u002Ftd\u003E\n\u003Ctd\u003EChromosomes to to be searched. If omitted all chromosomes will be prepared for searching.\u003C\u002Ftd\u003E\n\u003Ctd\u003Elist\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Estep_freq\u003C\u002Ftd\u003E\n\u003Ctd\u003EStep frequency of the sliding window approach. E.g., given an encoder with window size 12 kb, a step frequency of 6 means that every 2 kb a 12 kb window will be extracted from the bigWig.\u003C\u002Ftd\u003E\n\u003Ctd\u003Eint\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Edb_path\u003C\u002Ftd\u003E\n\u003Ctd\u003ERelative path to the sqlite db for storing searches.\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Ftbody\u003E\u003C\u002Ftable\u003E\n\u003Cp\u003EThe main parts to adjust are \u003Ccode\u003Eencoders\u003C\u002Fcode\u003E and \u003Ccode\u003Edatasets\u003C\u002Fcode\u003E. \u003Ccode\u003Eencoders\u003C\u002Fcode\u003E is a list of\n(auto)encoder definitions for different datatypes.T here are two ways to\nconfigure an (auto)encoder: (a) point to a pre-defined autoencoder or (b)\nconfigure from scratch.\u003C\u002Fp\u003E\n\u003Cp\u003EAssuming you want to use a predefined autoencoder all you have to do is\u003C\u002Fp\u003E\n\u003Ctable\u003E\n\u003Cthead\u003E\n\u003Ctr\u003E\n\u003Cth\u003EField\u003C\u002Fth\u003E\n\u003Cth\u003EDescription\u003C\u002Fth\u003E\n\u003Cth\u003EDtype\u003C\u002Fth\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Fthead\u003E\n\u003Ctbody\u003E\u003Ctr\u003E\n\u003Ctd\u003Econtent_type\u003C\u002Ftd\u003E\n\u003Ctd\u003EUnique string identifying the autoencoder in the configuration file\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Efrom_file\u003C\u002Ftd\u003E\n\u003Ctd\u003ERelative path to the encoder configuration file.\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Ftbody\u003E\u003C\u002Ftable\u003E\n\u003Cp\u003E\u003Cstrong\u003EExample:\u003C\u002Fstrong\u003E\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-json\"\u003E{\n  &quot;content_type&quot;: &quot;histone-mark-chip-seq-3kb&quot;,\n  &quot;from_file&quot;: &quot;examples\u002Fencoders.json&quot;\n}\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003EThe encoder configuration file is a dictionary with the top level keys acting\nas the identifier and need to match \u003Ccode\u003Econtent_type\u003C\u002Fcode\u003E above. Given the example\nfrom above the file could look like this:\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-json\"\u003E{\n  &quot;histone-mark-chip-seq-3kb&quot;: {},\n  &quot;dnase-seq-3kb&quot;: {}\n}\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003ESee \u003Ccode\u003E[encoders.json](encoders.json)\u003C\u002Fcode\u003E for an example. The specific definition if an\nautoencoder is the same as described in the following.\u003C\u002Fp\u003E\n\u003Cp\u003ETo configure an autoencoder from scratch you need to provide a dictionary with\nthe following required format:\u003C\u002Fp\u003E\n\u003Ctable\u003E\n\u003Cthead\u003E\n\u003Ctr\u003E\n\u003Cth\u003EField\u003C\u002Fth\u003E\n\u003Cth\u003EDescription\u003C\u002Fth\u003E\n\u003Cth\u003EDefaults\u003C\u002Fth\u003E\n\u003Cth\u003EDtype\u003C\u002Fth\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Fthead\u003E\n\u003Ctbody\u003E\u003Ctr\u003E\n\u003Ctd\u003Eautoencoder\u003C\u002Ftd\u003E\n\u003Ctd\u003ERelative path to your pickled autoencoder model. (hdf5 file)\u003C\u002Ftd\u003E\n\u003Ctd\u003E&nbsp;\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Eencoder\u003C\u002Ftd\u003E\n\u003Ctd\u003ERelative path to your pickled encoder model. (hdf5 file)\u003C\u002Ftd\u003E\n\u003Ctd\u003E&nbsp;\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Edecoder\u003C\u002Ftd\u003E\n\u003Ctd\u003ERelative path to your pickled decoder model. (hdf5 file)\u003C\u002Ftd\u003E\n\u003Ctd\u003E&nbsp;\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Econtent_type\u003C\u002Ftd\u003E\n\u003Ctd\u003EUnique string describing the content this autoencoder can handle. Data tracks with the same content type will be encoded by this autoencoder.\u003C\u002Ftd\u003E\n\u003Ctd\u003E&nbsp;\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Ewindow_size\u003C\u002Ftd\u003E\n\u003Ctd\u003EWindow size in base pairs used for training the autoencoder.\u003C\u002Ftd\u003E\n\u003Ctd\u003E&nbsp;\u003C\u002Ftd\u003E\n\u003Ctd\u003Eint\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Eresolution\u003C\u002Ftd\u003E\n\u003Ctd\u003EResolution or bin size of the window in base pairs.\u003C\u002Ftd\u003E\n\u003Ctd\u003E&nbsp;\u003C\u002Ftd\u003E\n\u003Ctd\u003Eint\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Elatent_dim\u003C\u002Ftd\u003E\n\u003Ctd\u003ENumber of latent dimensions of the encoded windows.\u003C\u002Ftd\u003E\n\u003Ctd\u003E&nbsp;\u003C\u002Ftd\u003E\n\u003Ctd\u003Eint\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Einput_dim\u003C\u002Ftd\u003E\n\u003Ctd\u003ENumber of input dimensions for Keras. For 1D data these are 3: samples, data length (which is \u003Ccode\u003Ewindow_size\u003C\u002Fcode\u003E \u002F \u003Ccode\u003Eresolution\u003C\u002Fcode\u003E), channels.\u003C\u002Ftd\u003E\n\u003Ctd\u003E3\u003C\u002Ftd\u003E\n\u003Ctd\u003Eint\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Echannels\u003C\u002Ftd\u003E\n\u003Ctd\u003ENumber of channels of the input data. This is normally 1.\u003C\u002Ftd\u003E\n\u003Ctd\u003E1\u003C\u002Ftd\u003E\n\u003Ctd\u003Eint\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Ftbody\u003E\u003C\u002Ftable\u003E\n\u003Cp\u003E\u003Cem\u003ENote that if you have specified an \u003Ccode\u003Eautoencoder\u003C\u002Fcode\u003E you do not need to provide\nseparate \u003Ccode\u003Eencoder\u003C\u002Fcode\u003E and \u003Ccode\u003Edecoder\u003C\u002Fcode\u003E models.\u003C\u002Fem\u003E\u003C\u002Fp\u003E\n\u003Cp\u003E\u003Cstrong\u003EExample:\u003C\u002Fstrong\u003E\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-json\"\u003E{\n  &quot;encoder&quot;: &quot;path\u002Fto\u002Fmy-12kb-chip-seq-encoder.h5&quot;,\n  &quot;decoder&quot;: &quot;path\u002Fto\u002Fmy-12kb-chip-seq-decoder.h5&quot;,\n  &quot;content_type&quot;: &quot;histone-mark-chip-seq&quot;,\n  &quot;window_size&quot;: 12000,\n  &quot;resolution&quot;: 100,\n  &quot;channels&quot;: 1,\n  &quot;input_dim&quot;: 3,\n  &quot;latent_dim&quot;: 12\n}\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003EDatasets require the following format:\u003C\u002Fp\u003E\n\u003Ctable\u003E\n\u003Cthead\u003E\n\u003Ctr\u003E\n\u003Cth\u003EField\u003C\u002Fth\u003E\n\u003Cth\u003EDescription\u003C\u002Fth\u003E\n\u003Cth\u003EDtype\u003C\u002Fth\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Fthead\u003E\n\u003Ctbody\u003E\u003Ctr\u003E\n\u003Ctd\u003Efilepath\u003C\u002Ftd\u003E\n\u003Ctd\u003ERelative path to the data file (bigWig or bigBed).\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Econtent_type\u003C\u002Ftd\u003E\n\u003Ctd\u003EUnique string describing the content this dataset. If you want to search for patterns in this track you need to have an autoencoder with a matching content type.\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Eid\u003C\u002Ftd\u003E\n\u003Ctd\u003EA unique string identifying your track. (Optional)\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003Ctr\u003E\n\u003Ctd\u003Ename\u003C\u002Ftd\u003E\n\u003Ctd\u003EA human readable name to be shown in HiGlass. (Optional)\u003C\u002Ftd\u003E\n\u003Ctd\u003Estr\u003C\u002Ftd\u003E\n\u003C\u002Ftr\u003E\n\u003C\u002Ftbody\u003E\u003C\u002Ftable\u003E\n\u003Cp\u003E\u003Cstrong\u003EExample:\u003C\u002Fstrong\u003E\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-json\"\u003E{\n  &quot;filepath&quot;: &quot;data\u002Fchip-seq\u002Fmy-fancy-gm12878-chip-seq-h3k27ac-fc-signal.bigWig&quot;,\n  &quot;content_type&quot;: &quot;histone-mark-chip-seq&quot;,\n  &quot;uuid&quot;: &quot;my-fancy-gm12878-chip-seq-h3k27c-track&quot;,\n  &quot;name&quot;: &quot;My Fancy GM12878 ChIP-Seq H3k27c Track&quot;\n}\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Ch4 id=\"start-peax\"\u003EStart Peax\u003C\u002Fh4\u003E\n\u003Cp\u003EFirst, start the Peax server to serve your data.\u003C\u002Fp\u003E\n\u003Cp\u003E\u003Cstrong\u003ENote:\u003C\u002Fstrong\u003E The first time you run Peax on a new dataset all the data will be prepared!\nDepending on your machine this can take some time. If you want to track the progress\nactivate the debugging mode using \u003Ccode\u003E-d\u003C\u002Fcode\u003E.\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-bash\"\u003Epython start.py\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003ENow go to \u003Ca href=\"http:\u002F\u002Flocalhost:5000\"\u003Ehttp:\u002F\u002Flocalhost:5000\u003C\u002Fa\u003E.\u003C\u002Fp\u003E\n\u003Cp\u003ETo \u003Ccode\u003Estart.py\u003C\u002Fcode\u003E script supports the following options:\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-bash\"\u003Eusage: start.py [-h] [-c CONFIG] [--clear] [--clear-cache]\n                [--clear-cache-at-exit] [--clear-db] [-d] [--host HOST]\n                [--port PORT] [-v]\n\nPeak Explorer CLI\n\noptional arguments:\n  -h, --help            show this help message and exit\n  -c CONFIG, --config CONFIG\n                        path to your JSON config file\n  --clear               clears the cache and database on startup\n  --clear-cache         clears the cache on startup\n  --clear-cache-at-exit\n                        clear the cache on shutdown\n  --clear-db            clears the database on startup\n  -d, --debug           turn on debug mode\n  --host HOST           customize the hostname\n  --port PORT           customize the port\n  -v, --verbose         turn verbose logging on\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003EThe \u003Ccode\u003Ehostname\u003C\u002Fcode\u003E defaults to \u003Ccode\u003Elocalhost\u003C\u002Fcode\u003E and the \u003Ccode\u003Eport\u003C\u002Fcode\u003E of the backend server defaults\nto \u003Ccode\u003E5000\u003C\u002Fcode\u003E.\u003C\u002Fp\u003E\n\u003Cp\u003EIn order to speed up subsequend user interaction, Peax initially prepapres all\nthe data and caches that data under \u003Ccode\u003E\u002Fcache\u003C\u002Fcode\u003E. You can always remove this\ndirectory manually or clear the cache on startup or at exist using the \u003Ccode\u003E--clear\u003C\u002Fcode\u003E\nas specified above.\u003C\u002Fp\u003E\n\u003Chr\u003E\n\u003Ch2 id=\"development\"\u003EDevelopment\u003C\u002Fh2\u003E\n\u003Cp\u003EHandy commands to keep in mind:\u003C\u002Fp\u003E\n\u003Cul\u003E\n\u003Cli\u003E\u003Ccode\u003Emake install\u003C\u002Fcode\u003E installs the conda environment and npm packages and builds the UI\u003C\u002Fli\u003E\n\u003Cli\u003E\u003Ccode\u003Emake update\u003C\u002Fcode\u003E updates the conda environment and npm packages and rebuilds the UI\u003C\u002Fli\u003E\n\u003Cli\u003E\u003Ccode\u003Emake build\u003C\u002Fcode\u003E builds the UI\u003C\u002Fli\u003E\n\u003Cli\u003E\u003Ccode\u003E.\u002Fstart.py\u003C\u002Fcode\u003E starts the Flask server application for serving data\u003C\u002Fli\u003E\n\u003Cli\u003E[\u002Fui]: \u003Ccode\u003Enpm install\u003C\u002Fcode\u003E installs and updates all the needed packages for the frontend\u003C\u002Fli\u003E\n\u003Cli\u003E[\u002Fui]: \u003Ccode\u003Enpm build\u003C\u002Fcode\u003E creates the production built of the frontend\u003C\u002Fli\u003E\n\u003Cli\u003E[\u002Fui]: \u003Ccode\u003Enpm start\u003C\u002Fcode\u003E starts a dev server with hot reloading for the frontend\u003C\u002Fli\u003E\n\u003C\u002Ful\u003E\n\u003Cp\u003ETo start developing on the server and the ui in parallel, first start the backend server\napplication using \u003Ccode\u003E.\u002Fstart.py\u003C\u002Fcode\u003E and then start the frontend server application from\n\u003Ccode\u003E.\u002Fui\u003C\u002Fcode\u003E using \u003Ccode\u003Enpm start\u003C\u002Fcode\u003E. Both server&#39;s watch the source code, so whenever you make\nchanges to the source code the servers will reload.\u003C\u002Fp\u003E\n\u003Ch3 id=\"configuration\"\u003EConfiguration\u003C\u002Fh3\u003E\n\u003Cp\u003EThere are 2 types of configuration files. The \u003Ca href=\"#configure-peax-with-your-data\"\u003Ebackend server configuration\u003C\u002Fa\u003E\ndefines the datasets to explore and is described in detail \u003Ca href=\"#configure-peax-with-your-data\"\u003Eabove\u003C\u002Fa\u003E.\u003C\u002Fp\u003E\n\u003Cp\u003EAdditionally, the frontend application can be configured to talk to a different backend\nserver and port if needed. Get started by copying the example configuration:\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-bash\"\u003Ecd ui &amp;&amp; cp config.json.sample config.json\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n\u003Cp\u003EBy default the \u003Ccode\u003Eserver\u003C\u002Fcode\u003E is dynamically set to the hostname of the server running the\nfrontend application. I.e., it is assumed that the backend server application is\nrunning on the same host as the frontend application. The \u003Ccode\u003Eport\u003C\u002Fcode\u003E of the server\ndefaults to \u003Ccode\u003E5000\u003C\u002Fcode\u003E.\u003C\u002Fp\u003E\n\u003Ch3 id=\"start-the-backend-and-frontend-apps\"\u003EStart the backend and frontend apps\u003C\u002Fh3\u003E\n\u003Cp\u003EFor development the backend and frontend applications run as seperate server\napplications.\u003C\u002Fp\u003E\n\u003Cpre\u003E\u003Ccode class=\"language-bash\"\u003E# Backend server\n.\u002Fstart.py --debug --config path\u002Fto\u002Fyour\u002Fconfig.json\n\n# Frontend server\ncd ui &amp;&amp; npm start\u003C\u002Fcode\u003E\u003C\u002Fpre\u003E\n"}}}("Peax","peax"))],session:{basepath:"",forArtifactory:false}};if('serviceWorker' in navigator)navigator.serviceWorker.register('/service-worker.js');</script><script src=/client/1ef9048e76cf74561a9e/main.js></script> 